// @lk/core/esm2022/lib/tools/carousel/carousel.component.mjs
import { Component, Input, Output, EventEmitter, ContentChild } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AppButtonComponent } from '../app-button/app-button.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
export class CarouselComponent {
    constructor() {
        this.items = [];
        this.height = '400px';
        this.width = '100%';
        this.autoPlay = false;
        this.autoPlayInterval = 3000;
        this.showIndicators = true;
        this.showControls = true;
        this.loop = true;
        this.itemChange = new EventEmitter();
        this.currentIndex = 0;
    }
    ngOnInit() {
        if (this.autoPlay && this.items.length > 1) {
            this.startAutoPlay();
        }
    }
    ngOnChanges(changes) {
        if (changes['items'] && this.items.length > 0) {
            if (this.currentIndex >= this.items.length) {
                this.currentIndex = 0;
            }
        }
        if (changes['autoPlay']) {
            if (this.autoPlay) {
                this.startAutoPlay();
            }
            else {
                this.stopAutoPlay();
            }
        }
    }
    ngOnDestroy() {
        this.stopAutoPlay();
    }
    startAutoPlay() {
        this.stopAutoPlay();
        if (this.autoPlay && this.items.length > 1) {
            this.autoPlayTimer = setInterval(() => {
                this.next();
            }, this.autoPlayInterval);
        }
    }
    stopAutoPlay() {
        if (this.autoPlayTimer) {
            clearInterval(this.autoPlayTimer);
            this.autoPlayTimer = null;
        }
    }
    next() {
        if (this.currentIndex < this.items.length - 1) {
            this.currentIndex++;
        }
        else if (this.loop) {
            this.currentIndex = 0;
        }
        this.emitItemChange();
    }
    previous() {
        if (this.currentIndex > 0) {
            this.currentIndex--;
        }
        else if (this.loop) {
            this.currentIndex = this.items.length - 1;
        }
        this.emitItemChange();
    }
    goTo(index) {
        if (index >= 0 && index < this.items.length) {
            this.currentIndex = index;
            this.emitItemChange();
        }
    }
    emitItemChange() {
        if (this.items[this.currentIndex]) {
            this.itemChange.emit({
                index: this.currentIndex,
                item: this.items[this.currentIndex]
            });
        }
    }
    get currentItem() {
        return this.items[this.currentIndex] || null;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: CarouselComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: CarouselComponent, isStandalone: true, selector: "app-carousel", inputs: { items: "items", height: "height", width: "width", autoPlay: "autoPlay", autoPlayInterval: "autoPlayInterval", showIndicators: "showIndicators", showControls: "showControls", loop: "loop" }, outputs: { itemChange: "itemChange" }, queries: [{ propertyName: "itemTemplate", first: true, predicate: ["carouselItemTemplate"], descendants: true }], usesOnChanges: true, ngImport: i0, template: "<div class=\"carousel-container\" [style.width]=\"width\" [style.height]=\"height\">\n  <div class=\"carousel-wrapper\">\n    @if (showControls && items.length > 1) {\n      <app-button\n        appearance=\"icon\"\n        type=\"button\"\n        [class]=\"'carousel-control prev'\"\n        [fontIcon]=\"'chevron_left'\"\n        (btnClick)=\"previous()\"\n      ></app-button>\n      <app-button\n        appearance=\"icon\"\n        type=\"button\"\n        [class]=\"'carousel-control next'\"\n        [fontIcon]=\"'chevron_right'\"\n        (btnClick)=\"next()\"\n      ></app-button>\n    }\n\n    <div class=\"carousel-content\">\n      @if (itemTemplate && currentItem) {\n        <ng-container *ngTemplateOutlet=\"itemTemplate; context: { $implicit: currentItem, index: currentIndex }\"></ng-container>\n      } @else if (currentItem) {\n        <div class=\"carousel-item-default\">\n          @if (currentItem.image) {\n            <img [src]=\"currentItem.image\" [alt]=\"currentItem.title || 'Carousel item'\" class=\"carousel-image\" />\n          }\n          @if (currentItem.title) {\n            <div class=\"carousel-title\">{{ currentItem.title }}</div>\n          }\n          @if (currentItem.description) {\n            <div class=\"carousel-description\">{{ currentItem.description }}</div>\n          }\n        </div>\n      }\n    </div>\n  </div>\n\n  @if (showIndicators && items.length > 1) {\n    <div class=\"carousel-indicators\">\n      @for (item of items; track $index) {\n        <app-button\n          appearance=\"icon\"\n          type=\"button\"\n          [class]=\"'indicator' + (currentIndex === $index ? ' active' : '')\"\n          (btnClick)=\"goTo($index)\"\n          [attr.aria-label]=\"'Go to slide ' + ($index + 1)\"\n        ></app-button>\n      }\n    </div>\n  }\n</div>\n\n", styles: [".carousel-container{position:relative;overflow:hidden;border-radius:4px}.carousel-wrapper{position:relative;width:100%;height:100%}.carousel-control{position:absolute;top:50%;transform:translateY(-50%);background-color:#00000080;color:#fff;border:none;border-radius:50%;width:40px;height:40px;display:flex;align-items:center;justify-content:center;cursor:pointer;z-index:10;transition:background-color .2s}.carousel-control:hover{background-color:#000000b3}.carousel-control mat-icon{font-size:24px;width:24px;height:24px}.carousel-control.prev{left:16px}.carousel-control.next{right:16px}.carousel-content{width:100%;height:100%;display:flex;align-items:center;justify-content:center}.carousel-item-default{width:100%;height:100%;display:flex;flex-direction:column;align-items:center;justify-content:center;text-align:center;padding:32px}.carousel-image{max-width:100%;max-height:100%;object-fit:contain;margin-bottom:16px}.carousel-title{font-size:24px;font-weight:500;margin-bottom:8px;color:#424242}.carousel-description{font-size:16px;color:#757575}.carousel-indicators{position:absolute;bottom:16px;left:50%;transform:translate(-50%);display:flex;gap:8px;z-index:10}.indicator{width:12px;height:12px;border-radius:50%;border:2px solid white;background-color:#ffffff80;cursor:pointer;transition:background-color .2s}.indicator:hover{background-color:#fffc}.indicator.active{background-color:#fff}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet", "ngTemplateOutletInjector"] }, { kind: "component", type: AppButtonComponent, selector: "app-button", inputs: ["color", "type", "disabled", "appearance", "class", "text", "fontIcon"], outputs: ["btnClick"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: CarouselComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-carousel', standalone: true, imports: [CommonModule, AppButtonComponent], template: "<div class=\"carousel-container\" [style.width]=\"width\" [style.height]=\"height\">\n  <div class=\"carousel-wrapper\">\n    @if (showControls && items.length > 1) {\n      <app-button\n        appearance=\"icon\"\n        type=\"button\"\n        [class]=\"'carousel-control prev'\"\n        [fontIcon]=\"'chevron_left'\"\n        (btnClick)=\"previous()\"\n      ></app-button>\n      <app-button\n        appearance=\"icon\"\n        type=\"button\"\n        [class]=\"'carousel-control next'\"\n        [fontIcon]=\"'chevron_right'\"\n        (btnClick)=\"next()\"\n      ></app-button>\n    }\n\n    <div class=\"carousel-content\">\n      @if (itemTemplate && currentItem) {\n        <ng-container *ngTemplateOutlet=\"itemTemplate; context: { $implicit: currentItem, index: currentIndex }\"></ng-container>\n      } @else if (currentItem) {\n        <div class=\"carousel-item-default\">\n          @if (currentItem.image) {\n            <img [src]=\"currentItem.image\" [alt]=\"currentItem.title || 'Carousel item'\" class=\"carousel-image\" />\n          }\n          @if (currentItem.title) {\n            <div class=\"carousel-title\">{{ currentItem.title }}</div>\n          }\n          @if (currentItem.description) {\n            <div class=\"carousel-description\">{{ currentItem.description }}</div>\n          }\n        </div>\n      }\n    </div>\n  </div>\n\n  @if (showIndicators && items.length > 1) {\n    <div class=\"carousel-indicators\">\n      @for (item of items; track $index) {\n        <app-button\n          appearance=\"icon\"\n          type=\"button\"\n          [class]=\"'indicator' + (currentIndex === $index ? ' active' : '')\"\n          (btnClick)=\"goTo($index)\"\n          [attr.aria-label]=\"'Go to slide ' + ($index + 1)\"\n        ></app-button>\n      }\n    </div>\n  }\n</div>\n\n", styles: [".carousel-container{position:relative;overflow:hidden;border-radius:4px}.carousel-wrapper{position:relative;width:100%;height:100%}.carousel-control{position:absolute;top:50%;transform:translateY(-50%);background-color:#00000080;color:#fff;border:none;border-radius:50%;width:40px;height:40px;display:flex;align-items:center;justify-content:center;cursor:pointer;z-index:10;transition:background-color .2s}.carousel-control:hover{background-color:#000000b3}.carousel-control mat-icon{font-size:24px;width:24px;height:24px}.carousel-control.prev{left:16px}.carousel-control.next{right:16px}.carousel-content{width:100%;height:100%;display:flex;align-items:center;justify-content:center}.carousel-item-default{width:100%;height:100%;display:flex;flex-direction:column;align-items:center;justify-content:center;text-align:center;padding:32px}.carousel-image{max-width:100%;max-height:100%;object-fit:contain;margin-bottom:16px}.carousel-title{font-size:24px;font-weight:500;margin-bottom:8px;color:#424242}.carousel-description{font-size:16px;color:#757575}.carousel-indicators{position:absolute;bottom:16px;left:50%;transform:translate(-50%);display:flex;gap:8px;z-index:10}.indicator{width:12px;height:12px;border-radius:50%;border:2px solid white;background-color:#ffffff80;cursor:pointer;transition:background-color .2s}.indicator:hover{background-color:#fffc}.indicator.active{background-color:#fff}\n"] }]
        }], propDecorators: { items: [{
                type: Input
            }], height: [{
                type: Input
            }], width: [{
                type: Input
            }], autoPlay: [{
                type: Input
            }], autoPlayInterval: [{
                type: Input
            }], showIndicators: [{
                type: Input
            }], showControls: [{
                type: Input
            }], loop: [{
                type: Input
            }], itemTemplate: [{
                type: ContentChild,
                args: ['carouselItemTemplate']
            }], itemChange: [{
                type: Output
            }] } });
