// @lk/core/esm2022/lib/guards/auth.guard.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/auth.service";
import * as i2 from "@angular/router";
export class AuthGuard {
    constructor(authService, router) {
        this.authService = authService;
        this.router = router;
    }
    canActivate() {
        // Check storage directly to avoid timing issues with state initialization
        const token = this.authService.getStoredToken();
        const user = this.authService.getStoredUser();
        const refreshToken = this.authService.getStoredRefreshToken();
        // Must have both token and user data to be considered authenticated
        if (!token || !user) {
            this.router.navigate(['/login']);
            return false;
        }
        // Check if token is valid
        const isTokenValid = this.authService.isTokenValid(token);
        if (isTokenValid) {
            return true;
        }
        // Token is expired - check if we have a refresh token
        if (refreshToken) {
            // Allow access - token refresh will happen in background via initializeAuthState
            return true;
        }
        // Token is expired and no refresh token
        // Don't redirect immediately - let initializeAuthState handle it
        // This prevents race conditions where guard runs before state is initialized
        // If state initialization clears data, the next navigation will redirect
        console.warn('Token expired and no refresh token found. Access may be denied after state initialization.');
        // Check auth state as fallback (in case initializeAuthState already ran)
        const isAuthenticated = this.authService.isAuthenticated();
        if (!isAuthenticated) {
            this.router.navigate(['/login']);
            return false;
        }
        // If authenticated in state but token expired and no refresh token, allow for now
        // The HTTP interceptor will handle 401 errors
        return true;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: AuthGuard, deps: [{ token: i1.AuthService }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: AuthGuard, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: AuthGuard, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.AuthService }, { type: i2.Router }] });
