// @lk/core/esm2022/lib/guards/login.guard.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../services/auth.service";
import * as i2 from "@angular/router";
export class LoginGuard {
    constructor(authService, router) {
        this.authService = authService;
        this.router = router;
    }
    canActivate() {
        // Check if user is authenticated and token is valid
        const isAuthenticated = this.authService.isAuthenticated();
        if (isAuthenticated && this.authService.isTokenValid()) {
            // User is already authenticated, redirect to appropriate dashboard
            const userType = this.authService.getUserType();
            if (userType === 'HOSPITAL') {
                this.router.navigate(['/admin-dashboard']);
            }
            else if (userType === 'DOCTOR') {
                this.router.navigate(['/dashboard']);
            }
            else if (userType === 'PATIENT') {
                this.router.navigate(['/patient-dashboard']);
            }
            else {
                // Default to dashboard
                this.router.navigate(['/dashboard']);
            }
            return false;
        }
        else {
            // Clear invalid auth data if exists
            if (isAuthenticated) {
                this.authService.logout();
            }
            // Allow access to login page
            return true;
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: LoginGuard, deps: [{ token: i1.AuthService }, { token: i2.Router }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: LoginGuard, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: LoginGuard, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.AuthService }, { type: i2.Router }] });
