// @lk/core/esm2022/lib/tools/radio-button/radio-button.component.mjs
import { Component, Input, Output, EventEmitter, forwardRef } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, NG_VALUE_ACCESSOR } from '@angular/forms';
import { MatRadioModule } from '@angular/material/radio';
import { MatFormFieldModule } from '@angular/material/form-field';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/forms";
import * as i3 from "@angular/material/radio";
export class RadioButtonComponent {
    constructor() {
        this.options = [];
        this.value = null;
        this.label = '';
        this.name = '';
        this.color = 'primary';
        this.disabled = false;
        this.vertical = false;
        this.labelTemplateRef = null;
        this.valueChange = new EventEmitter();
        this.onChange = (value) => { };
        this.onTouched = () => { };
    }
    writeValue(value) {
        this.value = value;
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    onSelectionChange(value) {
        this.value = value;
        this.onChange(value);
        this.onTouched();
        this.valueChange.emit(value);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: RadioButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: RadioButtonComponent, isStandalone: true, selector: "app-radio-button", inputs: { options: "options", value: "value", label: "label", name: "name", color: "color", disabled: "disabled", vertical: "vertical", labelTemplateRef: "labelTemplateRef" }, outputs: { valueChange: "valueChange" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(() => RadioButtonComponent),
                multi: true
            }
        ], ngImport: i0, template: "<div class=\"radio-button-container\" [class.vertical]=\"vertical\">\n  @if (labelTemplateRef) {\n    <ng-container *ngTemplateOutlet=\"labelTemplateRef; context: { $implicit: label, label: label }\"></ng-container>\n  } @else if (label) {\n    <label class=\"radio-group-label\">{{ label }}</label>\n  }\n  <mat-radio-group\n    [(ngModel)]=\"value\"\n    [name]=\"name\"\n    [color]=\"color\"\n    [disabled]=\"disabled\"\n    (change)=\"onSelectionChange($event.value)\"\n    class=\"radio-group\"\n    [class.vertical]=\"vertical\"\n  >\n    @for (option of options; track option.value) {\n      <mat-radio-button [value]=\"option.value\" [disabled]=\"option.disabled || disabled\">\n        {{ option.label }}\n      </mat-radio-button>\n    }\n  </mat-radio-group>\n</div>\n\n", styles: [".radio-button-container{display:flex;flex-direction:column;gap:8px}.radio-group-label{font-size:14px;font-weight:500;color:#424242}.radio-group{display:flex;gap:16px;flex-wrap:wrap}.radio-group.vertical{flex-direction:column;gap:8px}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet", "ngTemplateOutletInjector"] }, { kind: "ngmodule", type: FormsModule }, { kind: "directive", type: i2.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i2.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }, { kind: "ngmodule", type: MatRadioModule }, { kind: "directive", type: i3.MatRadioGroup, selector: "mat-radio-group", inputs: ["color", "name", "labelPosition", "value", "selected", "disabled", "required"], outputs: ["change"], exportAs: ["matRadioGroup"] }, { kind: "component", type: i3.MatRadioButton, selector: "mat-radio-button", inputs: ["id", "name", "aria-label", "aria-labelledby", "aria-describedby", "disableRipple", "tabIndex", "checked", "value", "labelPosition", "disabled", "required", "color"], outputs: ["change"], exportAs: ["matRadioButton"] }, { kind: "ngmodule", type: MatFormFieldModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: RadioButtonComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-radio-button', standalone: true, imports: [CommonModule, FormsModule, MatRadioModule, MatFormFieldModule], providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef(() => RadioButtonComponent),
                            multi: true
                        }
                    ], template: "<div class=\"radio-button-container\" [class.vertical]=\"vertical\">\n  @if (labelTemplateRef) {\n    <ng-container *ngTemplateOutlet=\"labelTemplateRef; context: { $implicit: label, label: label }\"></ng-container>\n  } @else if (label) {\n    <label class=\"radio-group-label\">{{ label }}</label>\n  }\n  <mat-radio-group\n    [(ngModel)]=\"value\"\n    [name]=\"name\"\n    [color]=\"color\"\n    [disabled]=\"disabled\"\n    (change)=\"onSelectionChange($event.value)\"\n    class=\"radio-group\"\n    [class.vertical]=\"vertical\"\n  >\n    @for (option of options; track option.value) {\n      <mat-radio-button [value]=\"option.value\" [disabled]=\"option.disabled || disabled\">\n        {{ option.label }}\n      </mat-radio-button>\n    }\n  </mat-radio-group>\n</div>\n\n", styles: [".radio-button-container{display:flex;flex-direction:column;gap:8px}.radio-group-label{font-size:14px;font-weight:500;color:#424242}.radio-group{display:flex;gap:16px;flex-wrap:wrap}.radio-group.vertical{flex-direction:column;gap:8px}\n"] }]
        }], propDecorators: { options: [{
                type: Input
            }], value: [{
                type: Input
            }], label: [{
                type: Input
            }], name: [{
                type: Input
            }], color: [{
                type: Input
            }], disabled: [{
                type: Input
            }], vertical: [{
                type: Input
            }], labelTemplateRef: [{
                type: Input
            }], valueChange: [{
                type: Output
            }] } });
