// @lk/core/esm2022/lib/tools/date-picker/date-picker.component.mjs
import { Component, Input, forwardRef } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NG_VALUE_ACCESSOR, FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatDatepickerModule } from '@angular/material/datepicker';
import { MatInputModule } from '@angular/material/input';
import { MatNativeDateModule } from '@angular/material/core';
import { MatIconModule } from '@angular/material/icon';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/material/form-field";
import * as i3 from "@angular/material/datepicker";
import * as i4 from "@angular/material/input";
export class DatePickerComponent {
    constructor() {
        this.label = '';
        this.placeholder = 'Select date';
        this.required = false;
        this.disabled = false;
        this.minDate = null;
        this.maxDate = null;
        this.startView = 'month';
        this.value = null;
        this.touched = false;
        // ControlValueAccessor methods
        this.onChange = (value) => { };
        this.onTouched = () => { };
    }
    ngOnInit() {
        // Convert string inputs to Date objects if needed
        if (this.minDate && typeof this.minDate === 'string') {
            this.minDate = new Date(this.minDate);
        }
        if (this.maxDate && typeof this.maxDate === 'string') {
            this.maxDate = new Date(this.maxDate);
        }
    }
    writeValue(value) {
        this.value = value;
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
    onDateChange(date) {
        this.value = date;
        this.onChange(date);
        this.markAsTouched();
    }
    onDateInput(event) {
        const date = event.target.value ? new Date(event.target.value) : null;
        this.onDateChange(date);
    }
    markAsTouched() {
        if (!this.touched) {
            this.touched = true;
            this.onTouched();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DatePickerComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: DatePickerComponent, isStandalone: true, selector: "app-date-picker", inputs: { label: "label", placeholder: "placeholder", required: "required", disabled: "disabled", minDate: "minDate", maxDate: "maxDate", startView: "startView" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(() => DatePickerComponent),
                multi: true
            }
        ], ngImport: i0, template: "<mat-form-field appearance=\"outline\" class=\"date-picker-field\">\n  <mat-label *ngIf=\"label\">{{ label }}</mat-label>\n  <input \n    matInput \n    [matDatepicker]=\"picker\" \n    [placeholder]=\"placeholder\"\n    [required]=\"required\"\n    [disabled]=\"disabled\"\n    [min]=\"minDate\"\n    [max]=\"maxDate\"\n    [value]=\"value\"\n    (dateChange)=\"onDateChange($event.value)\"\n    (input)=\"onDateInput($event)\"\n    (blur)=\"markAsTouched()\">\n  <mat-datepicker-toggle matSuffix [for]=\"picker\"></mat-datepicker-toggle>\n  <mat-datepicker #picker [startView]=\"startView\"></mat-datepicker>\n  <mat-error *ngIf=\"required && !value\">\n    This field is required\n  </mat-error>\n</mat-form-field> ", styles: [".date-picker-field{width:100%}.date-picker-field ::ng-deep .mat-mdc-form-field{width:100%}.date-picker-field ::ng-deep .mat-mdc-text-field-wrapper{background-color:#fff;border-radius:8px;border:1px solid #e2e8f0;transition:border-color .2s ease,box-shadow .2s ease}.date-picker-field ::ng-deep .mat-mdc-text-field-wrapper:hover{border-color:#10b3b3}.date-picker-field ::ng-deep .mat-mdc-text-field-wrapper.mat-focused{border-color:#10b3b3;box-shadow:0 0 0 3px #10b3b31a}.date-picker-field ::ng-deep .mat-mdc-form-field-focus-overlay{background-color:transparent}.date-picker-field ::ng-deep .mat-mdc-form-field-subscript-wrapper{display:none}.date-picker-field ::ng-deep .mat-mdc-form-field-infix{padding:12px 0;min-height:auto}.date-picker-field ::ng-deep .mat-mdc-input-element{font-size:14px;color:#374151}.date-picker-field ::ng-deep .mat-mdc-input-element::placeholder{color:#9ca3af}.date-picker-field ::ng-deep .mat-mdc-input-element:disabled{color:#9ca3af;background-color:#f9fafb}.date-picker-field ::ng-deep .mat-mdc-form-field-label{color:#6b7280;font-size:14px}.date-picker-field ::ng-deep .mat-mdc-form-field-label.mat-mdc-form-field-label-alternate{color:#10b3b3}.date-picker-field ::ng-deep .mat-mdc-datepicker-toggle{color:#6b7280}.date-picker-field ::ng-deep .mat-mdc-datepicker-toggle:hover{color:#10b3b3}.date-picker-field ::ng-deep .mat-mdc-form-field-error{color:#ef4444;font-size:12px;margin-top:4px}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "ngmodule", type: FormsModule }, { kind: "ngmodule", type: ReactiveFormsModule }, { kind: "ngmodule", type: MatFormFieldModule }, { kind: "component", type: i2.MatFormField, selector: "mat-form-field", inputs: ["hideRequiredMarker", "color", "floatLabel", "appearance", "subscriptSizing", "hintLabel"], exportAs: ["matFormField"] }, { kind: "directive", type: i2.MatLabel, selector: "mat-label" }, { kind: "directive", type: i2.MatError, selector: "mat-error, [matError]", inputs: ["id"] }, { kind: "directive", type: i2.MatSuffix, selector: "[matSuffix], [matIconSuffix], [matTextSuffix]", inputs: ["matTextSuffix"] }, { kind: "ngmodule", type: MatDatepickerModule }, { kind: "component", type: i3.MatDatepicker, selector: "mat-datepicker", exportAs: ["matDatepicker"] }, { kind: "directive", type: i3.MatDatepickerInput, selector: "input[matDatepicker]", inputs: ["matDatepicker", "min", "max", "matDatepickerFilter"], exportAs: ["matDatepickerInput"] }, { kind: "component", type: i3.MatDatepickerToggle, selector: "mat-datepicker-toggle", inputs: ["for", "tabIndex", "aria-label", "disabled", "disableRipple"], exportAs: ["matDatepickerToggle"] }, { kind: "ngmodule", type: MatInputModule }, { kind: "directive", type: i4.MatInput, selector: "input[matInput], textarea[matInput], select[matNativeControl],      input[matNativeControl], textarea[matNativeControl]", inputs: ["disabled", "id", "placeholder", "name", "required", "type", "errorStateMatcher", "aria-describedby", "value", "readonly"], exportAs: ["matInput"] }, { kind: "ngmodule", type: MatNativeDateModule }, { kind: "ngmodule", type: MatIconModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: DatePickerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-date-picker', standalone: true, imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        MatFormFieldModule,
                        MatDatepickerModule,
                        MatInputModule,
                        MatNativeDateModule,
                        MatIconModule
                    ], providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef(() => DatePickerComponent),
                            multi: true
                        }
                    ], template: "<mat-form-field appearance=\"outline\" class=\"date-picker-field\">\n  <mat-label *ngIf=\"label\">{{ label }}</mat-label>\n  <input \n    matInput \n    [matDatepicker]=\"picker\" \n    [placeholder]=\"placeholder\"\n    [required]=\"required\"\n    [disabled]=\"disabled\"\n    [min]=\"minDate\"\n    [max]=\"maxDate\"\n    [value]=\"value\"\n    (dateChange)=\"onDateChange($event.value)\"\n    (input)=\"onDateInput($event)\"\n    (blur)=\"markAsTouched()\">\n  <mat-datepicker-toggle matSuffix [for]=\"picker\"></mat-datepicker-toggle>\n  <mat-datepicker #picker [startView]=\"startView\"></mat-datepicker>\n  <mat-error *ngIf=\"required && !value\">\n    This field is required\n  </mat-error>\n</mat-form-field> ", styles: [".date-picker-field{width:100%}.date-picker-field ::ng-deep .mat-mdc-form-field{width:100%}.date-picker-field ::ng-deep .mat-mdc-text-field-wrapper{background-color:#fff;border-radius:8px;border:1px solid #e2e8f0;transition:border-color .2s ease,box-shadow .2s ease}.date-picker-field ::ng-deep .mat-mdc-text-field-wrapper:hover{border-color:#10b3b3}.date-picker-field ::ng-deep .mat-mdc-text-field-wrapper.mat-focused{border-color:#10b3b3;box-shadow:0 0 0 3px #10b3b31a}.date-picker-field ::ng-deep .mat-mdc-form-field-focus-overlay{background-color:transparent}.date-picker-field ::ng-deep .mat-mdc-form-field-subscript-wrapper{display:none}.date-picker-field ::ng-deep .mat-mdc-form-field-infix{padding:12px 0;min-height:auto}.date-picker-field ::ng-deep .mat-mdc-input-element{font-size:14px;color:#374151}.date-picker-field ::ng-deep .mat-mdc-input-element::placeholder{color:#9ca3af}.date-picker-field ::ng-deep .mat-mdc-input-element:disabled{color:#9ca3af;background-color:#f9fafb}.date-picker-field ::ng-deep .mat-mdc-form-field-label{color:#6b7280;font-size:14px}.date-picker-field ::ng-deep .mat-mdc-form-field-label.mat-mdc-form-field-label-alternate{color:#10b3b3}.date-picker-field ::ng-deep .mat-mdc-datepicker-toggle{color:#6b7280}.date-picker-field ::ng-deep .mat-mdc-datepicker-toggle:hover{color:#10b3b3}.date-picker-field ::ng-deep .mat-mdc-form-field-error{color:#ef4444;font-size:12px;margin-top:4px}\n"] }]
        }], propDecorators: { label: [{
                type: Input
            }], placeholder: [{
                type: Input
            }], required: [{
                type: Input
            }], disabled: [{
                type: Input
            }], minDate: [{
                type: Input
            }], maxDate: [{
                type: Input
            }], startView: [{
                type: Input
            }] } });
