// @lk/core/esm2022/lib/tools/chip-cell-renderer/chip-cell-renderer.component.mjs
import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CHIP_COLORS } from '../../constants/chip.constant';
import * as i0 from "@angular/core";
export class ChipCellRendererComponent {
    constructor() {
        this.params = {};
        this.status = '';
        this.color = '';
        this.chipColors = CHIP_COLORS;
    }
    agInit(params) {
        this.params = params;
        this.status = params.value;
        this.color = this.chipColors[this.status] ?? 'gray';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ChipCellRendererComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: ChipCellRendererComponent, isStandalone: true, selector: "app-chip-cell-renderer", ngImport: i0, template: "<div [style.background-color]=\"color\" class=\"status-chip\">{{ status }}</div> ", styles: [".status-chip{font-weight:500;font-size:1em;justify-content:center;border-radius:4em;padding:.5em 1em;color:#fff}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ChipCellRendererComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-chip-cell-renderer', standalone: true, imports: [CommonModule], template: "<div [style.background-color]=\"color\" class=\"status-chip\">{{ status }}</div> ", styles: [".status-chip{font-weight:500;font-size:1em;justify-content:center;border-radius:4em;padding:.5em 1em;color:#fff}\n"] }]
        }] });
