// @lk/core/esm2022/lib/tools/qr-code-generator/qr-code-generator.component.mjs
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatButtonModule } from '@angular/material/button';
import { MatInputModule } from '@angular/material/input';
import { MatFormFieldModule } from '@angular/material/form-field';
import { FormsModule } from '@angular/forms';
import { AppButtonComponent } from '../app-button/app-button.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/input";
import * as i2 from "@angular/material/form-field";
import * as i3 from "@angular/forms";
export class QrCodeGeneratorComponent {
    constructor() {
        this.data = '';
        this.size = 200;
        this.errorCorrectionLevel = 'M';
        this.showInput = true;
        this.showDownload = true;
        this.qrGenerated = new EventEmitter();
        this.qrCodeDataUrl = '';
        this.inputData = '';
    }
    ngOnInit() {
        this.inputData = this.data;
        if (this.data) {
            this.generateQRCode();
        }
    }
    ngOnChanges(changes) {
        if (changes['data'] && this.data) {
            this.inputData = this.data;
            this.generateQRCode();
        }
    }
    generateQRCode() {
        if (!this.inputData) {
            this.qrCodeDataUrl = '';
            return;
        }
        // Using a simple QR code generation approach
        // In production, you would use a library like 'qrcode' or 'angularx-qrcode'
        // For now, we'll use a QR code API service
        const qrApiUrl = `https://api.qrserver.com/v1/create-qr-code/?size=${this.size}x${this.size}&data=${encodeURIComponent(this.inputData)}&ecc=${this.errorCorrectionLevel}`;
        this.qrCodeDataUrl = qrApiUrl;
        this.qrGenerated.emit(this.qrCodeDataUrl);
    }
    onInputChange() {
        this.generateQRCode();
    }
    downloadQRCode() {
        if (this.qrCodeDataUrl) {
            const link = document.createElement('a');
            link.href = this.qrCodeDataUrl;
            link.download = 'qrcode.png';
            link.click();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: QrCodeGeneratorComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: QrCodeGeneratorComponent, isStandalone: true, selector: "app-qr-code-generator", inputs: { data: "data", size: "size", errorCorrectionLevel: "errorCorrectionLevel", showInput: "showInput", showDownload: "showDownload" }, outputs: { qrGenerated: "qrGenerated" }, usesOnChanges: true, ngImport: i0, template: "<div class=\"qr-code-container\">\n  @if (showInput) {\n    <div class=\"qr-input-section\">\n      <mat-form-field appearance=\"outline\" class=\"full-width\">\n        <mat-label>Enter data for QR code</mat-label>\n        <input\n          matInput\n          [(ngModel)]=\"inputData\"\n          (ngModelChange)=\"onInputChange()\"\n          placeholder=\"Enter text or URL\"\n        />\n      </mat-form-field>\n    </div>\n  }\n\n  @if (qrCodeDataUrl) {\n    <div class=\"qr-code-display\">\n      <img [src]=\"qrCodeDataUrl\" [style.width.px]=\"size\" [style.height.px]=\"size\" alt=\"QR Code\" />\n    </div>\n  } @else {\n    <div class=\"qr-placeholder\">\n      <p>Enter data to generate QR code</p>\n    </div>\n  }\n\n  @if (showDownload && qrCodeDataUrl) {\n    <div class=\"qr-actions\">\n      <app-button\n        text=\"Download\"\n        fontIcon=\"download\"\n        appearance=\"raised\"\n        color=\"primary\"\n        (btnClick)=\"downloadQRCode()\"\n      ></app-button>\n    </div>\n  }\n</div>\n\n", styles: [".qr-code-container{display:flex;flex-direction:column;align-items:center;gap:16px;padding:16px}.qr-input-section{width:100%;max-width:400px}.full-width{width:100%}.qr-code-display{display:flex;justify-content:center;align-items:center;padding:16px;background-color:#fff;border:1px solid #e0e0e0;border-radius:4px}.qr-placeholder{padding:32px;text-align:center;color:#9e9e9e}.qr-actions{display:flex;gap:8px}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatButtonModule }, { kind: "ngmodule", type: MatInputModule }, { kind: "directive", type: i1.MatInput, selector: "input[matInput], textarea[matInput], select[matNativeControl],      input[matNativeControl], textarea[matNativeControl]", inputs: ["disabled", "id", "placeholder", "name", "required", "type", "errorStateMatcher", "aria-describedby", "value", "readonly"], exportAs: ["matInput"] }, { kind: "component", type: i2.MatFormField, selector: "mat-form-field", inputs: ["hideRequiredMarker", "color", "floatLabel", "appearance", "subscriptSizing", "hintLabel"], exportAs: ["matFormField"] }, { kind: "directive", type: i2.MatLabel, selector: "mat-label" }, { kind: "ngmodule", type: MatFormFieldModule }, { kind: "ngmodule", type: FormsModule }, { kind: "directive", type: i3.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i3.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i3.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }, { kind: "component", type: AppButtonComponent, selector: "app-button", inputs: ["color", "type", "disabled", "appearance", "class", "text", "fontIcon"], outputs: ["btnClick"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: QrCodeGeneratorComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-qr-code-generator', standalone: true, imports: [
                        CommonModule,
                        MatButtonModule,
                        MatInputModule,
                        MatFormFieldModule,
                        FormsModule,
                        AppButtonComponent
                    ], template: "<div class=\"qr-code-container\">\n  @if (showInput) {\n    <div class=\"qr-input-section\">\n      <mat-form-field appearance=\"outline\" class=\"full-width\">\n        <mat-label>Enter data for QR code</mat-label>\n        <input\n          matInput\n          [(ngModel)]=\"inputData\"\n          (ngModelChange)=\"onInputChange()\"\n          placeholder=\"Enter text or URL\"\n        />\n      </mat-form-field>\n    </div>\n  }\n\n  @if (qrCodeDataUrl) {\n    <div class=\"qr-code-display\">\n      <img [src]=\"qrCodeDataUrl\" [style.width.px]=\"size\" [style.height.px]=\"size\" alt=\"QR Code\" />\n    </div>\n  } @else {\n    <div class=\"qr-placeholder\">\n      <p>Enter data to generate QR code</p>\n    </div>\n  }\n\n  @if (showDownload && qrCodeDataUrl) {\n    <div class=\"qr-actions\">\n      <app-button\n        text=\"Download\"\n        fontIcon=\"download\"\n        appearance=\"raised\"\n        color=\"primary\"\n        (btnClick)=\"downloadQRCode()\"\n      ></app-button>\n    </div>\n  }\n</div>\n\n", styles: [".qr-code-container{display:flex;flex-direction:column;align-items:center;gap:16px;padding:16px}.qr-input-section{width:100%;max-width:400px}.full-width{width:100%}.qr-code-display{display:flex;justify-content:center;align-items:center;padding:16px;background-color:#fff;border:1px solid #e0e0e0;border-radius:4px}.qr-placeholder{padding:32px;text-align:center;color:#9e9e9e}.qr-actions{display:flex;gap:8px}\n"] }]
        }], propDecorators: { data: [{
                type: Input
            }], size: [{
                type: Input
            }], errorCorrectionLevel: [{
                type: Input
            }], showInput: [{
                type: Input
            }], showDownload: [{
                type: Input
            }], qrGenerated: [{
                type: Output
            }] } });
