// @lk/core/esm2022/lib/tools/grid-list-filters/grid-list-filters.component.mjs
import { Component, Input, Output, EventEmitter, inject } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatDialog } from '@angular/material/dialog';
import { AppButtonComponent } from '../app-button/app-button.component';
import { FilterComponent } from '../filter/filter.component';
import { DialogboxService } from '../dialogbox/dialogbox.component';
import * as i0 from "@angular/core";
export class GridListFiltersComponent {
    constructor() {
        this.fields = [];
        this.operators = [];
        this.operatorsMap = {};
        this.valuesMap = {};
        this.title = 'Filter Data';
        this.buttonText = 'Filter';
        this.buttonIcon = 'filter_list';
        this.filtersApplied = new EventEmitter();
        this.filtersCleared = new EventEmitter();
        this.dialog = inject(MatDialog);
        this.dialogboxService = inject(DialogboxService);
        this.appliedFiltersCount = 0;
    }
    openFiltersDialog() {
        const dialogRef = this.dialogboxService.openDialog(FilterComponent, {
            title: this.title,
            width: '600px',
            data: {
                fields: this.fields,
                operators: this.operators,
                operatorsMap: this.operatorsMap,
                valuesMap: this.valuesMap,
                title: this.title
            }
        });
        dialogRef.afterClosed().subscribe((result) => {
            if (result && result.filters) {
                this.appliedFiltersCount = result.filters.length;
                this.filtersApplied.emit(result.filters);
            }
        });
    }
    clearFilters() {
        this.appliedFiltersCount = 0;
        this.filtersCleared.emit();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: GridListFiltersComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: GridListFiltersComponent, isStandalone: true, selector: "app-grid-list-filters", inputs: { fields: "fields", operators: "operators", operatorsMap: "operatorsMap", valuesMap: "valuesMap", title: "title", buttonText: "buttonText", buttonIcon: "buttonIcon" }, outputs: { filtersApplied: "filtersApplied", filtersCleared: "filtersCleared" }, ngImport: i0, template: "<div class=\"grid-filters-container\">\n  <app-button\n    [text]=\"buttonText\"\n    [fontIcon]=\"buttonIcon\"\n    appearance=\"raised\"\n    color=\"primary\"\n    (btnClick)=\"openFiltersDialog()\"\n    [class]=\"'filter-button'\"\n  ></app-button>\n  \n  @if (appliedFiltersCount > 0) {\n    <span class=\"filter-badge\">{{ appliedFiltersCount }}</span>\n    <app-button\n      text=\"Clear\"\n      fontIcon=\"clear\"\n      appearance=\"flat\"\n      color=\"warn\"\n      (btnClick)=\"clearFilters()\"\n      [class]=\"'clear-button'\"\n    ></app-button>\n  }\n</div>\n\n", styles: [".grid-filters-container{display:flex;align-items:center;gap:8px}.filter-button{margin-right:8px}.filter-badge{display:inline-flex;align-items:center;justify-content:center;min-width:20px;height:20px;padding:0 6px;background-color:#f44336;color:#fff;border-radius:10px;font-size:12px;font-weight:500}.clear-button{margin-left:4px}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatButtonModule }, { kind: "ngmodule", type: MatIconModule }, { kind: "component", type: AppButtonComponent, selector: "app-button", inputs: ["color", "type", "disabled", "appearance", "class", "text", "fontIcon"], outputs: ["btnClick"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: GridListFiltersComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-grid-list-filters', standalone: true, imports: [CommonModule, MatButtonModule, MatIconModule, AppButtonComponent], template: "<div class=\"grid-filters-container\">\n  <app-button\n    [text]=\"buttonText\"\n    [fontIcon]=\"buttonIcon\"\n    appearance=\"raised\"\n    color=\"primary\"\n    (btnClick)=\"openFiltersDialog()\"\n    [class]=\"'filter-button'\"\n  ></app-button>\n  \n  @if (appliedFiltersCount > 0) {\n    <span class=\"filter-badge\">{{ appliedFiltersCount }}</span>\n    <app-button\n      text=\"Clear\"\n      fontIcon=\"clear\"\n      appearance=\"flat\"\n      color=\"warn\"\n      (btnClick)=\"clearFilters()\"\n      [class]=\"'clear-button'\"\n    ></app-button>\n  }\n</div>\n\n", styles: [".grid-filters-container{display:flex;align-items:center;gap:8px}.filter-button{margin-right:8px}.filter-badge{display:inline-flex;align-items:center;justify-content:center;min-width:20px;height:20px;padding:0 6px;background-color:#f44336;color:#fff;border-radius:10px;font-size:12px;font-weight:500}.clear-button{margin-left:4px}\n"] }]
        }], propDecorators: { fields: [{
                type: Input
            }], operators: [{
                type: Input
            }], operatorsMap: [{
                type: Input
            }], valuesMap: [{
                type: Input
            }], title: [{
                type: Input
            }], buttonText: [{
                type: Input
            }], buttonIcon: [{
                type: Input
            }], filtersApplied: [{
                type: Output
            }], filtersCleared: [{
                type: Output
            }] } });
