// @lk/core/esm2022/lib/tools/confirmation-dialog/confirmation-dialog.component.mjs
import { Component, inject } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DIALOG_DATA_TOKEN } from '../index';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
export class ConfirmationDialogComponent {
    constructor() {
        this.data = inject(DIALOG_DATA_TOKEN);
        this.destroy$ = new Subject();
    }
    ngOnInit() {
        // Component just displays the message
        // Footer actions are handled by DialogboxContentComponent
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    get displayMessage() {
        return this.data?.message || 'Are you sure?';
    }
    get displayTitle() {
        return this.data?.title || 'Confirm';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ConfirmationDialogComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: ConfirmationDialogComponent, isStandalone: true, selector: "app-confirmation-dialog", ngImport: i0, template: "<div class=\"confirmation-dialog\">\n  <div class=\"confirmation-content\">\n    <div class=\"message-container\">\n      <p class=\"confirmation-message\">{{ displayMessage }}</p>\n    </div>\n  </div>\n</div>\n\n", styles: [".confirmation-dialog{padding:0}.confirmation-dialog .confirmation-content{display:flex;flex-direction:column}.confirmation-dialog .confirmation-content .message-container{width:100%}.confirmation-dialog .confirmation-content .message-container .confirmation-message{margin:0;font-size:16px;line-height:1.6;color:#1f2937;word-wrap:break-word;text-align:left}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ConfirmationDialogComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-confirmation-dialog', standalone: true, imports: [
                        CommonModule
                    ], template: "<div class=\"confirmation-dialog\">\n  <div class=\"confirmation-content\">\n    <div class=\"message-container\">\n      <p class=\"confirmation-message\">{{ displayMessage }}</p>\n    </div>\n  </div>\n</div>\n\n", styles: [".confirmation-dialog{padding:0}.confirmation-dialog .confirmation-content{display:flex;flex-direction:column}.confirmation-dialog .confirmation-content .message-container{width:100%}.confirmation-dialog .confirmation-content .message-container .confirmation-message{margin:0;font-size:16px;line-height:1.6;color:#1f2937;word-wrap:break-word;text-align:left}\n"] }]
        }], ctorParameters: () => [] });
