// @lk/core/esm2022/lib/tools/toggle-button/toggle-button.component.mjs
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatButtonToggleModule } from '@angular/material/button-toggle';
import { MatButtonModule } from '@angular/material/button';
import { AppButtonComponent } from '../app-button/app-button.component';
import { IconComponent } from '../app-icon/icon.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/button-toggle";
export class ToggleButtonComponent {
    constructor() {
        this.options = [];
        this.value = null;
        this.multiple = false;
        this.appearance = 'standard';
        this.vertical = false;
        this.valueChange = new EventEmitter();
        this.buttonClick = new EventEmitter();
        this.selectedValues = [];
    }
    ngOnInit() {
        if (this.multiple) {
            this.selectedValues = Array.isArray(this.value) ? [...this.value] : [];
        }
    }
    onToggleChange(value) {
        if (this.multiple) {
            this.selectedValues = Array.isArray(value) ? value : [value];
            this.valueChange.emit(this.selectedValues);
        }
        else {
            this.value = value;
            this.valueChange.emit(value);
            const option = this.options.find(opt => opt.value === value);
            if (option) {
                this.buttonClick.emit({ value, option });
            }
        }
    }
    onClick(option) {
        if (!option.disabled) {
            if (this.multiple) {
                const index = this.selectedValues.indexOf(option.value);
                if (index > -1) {
                    this.selectedValues.splice(index, 1);
                }
                else {
                    this.selectedValues.push(option.value);
                }
                this.valueChange.emit([...this.selectedValues]);
                this.buttonClick.emit({ value: option.value, option });
            }
            else {
                this.value = option.value;
                this.valueChange.emit(option.value);
                this.buttonClick.emit({ value: option.value, option });
            }
        }
    }
    isSelected(option) {
        if (this.multiple) {
            return this.selectedValues.includes(option.value);
        }
        return this.value === option.value;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ToggleButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: ToggleButtonComponent, isStandalone: true, selector: "app-toggle-button", inputs: { options: "options", value: "value", multiple: "multiple", appearance: "appearance", vertical: "vertical" }, outputs: { valueChange: "valueChange", buttonClick: "buttonClick" }, ngImport: i0, template: "@if (multiple) {\n  <mat-button-toggle-group\n    [value]=\"selectedValues\"\n    [multiple]=\"true\"\n    [appearance]=\"appearance\"\n    [vertical]=\"vertical\"\n    (change)=\"onToggleChange($event.value)\"\n    class=\"toggle-group\"\n  >\n    @for (option of options; track option.value) {\n      <mat-button-toggle [value]=\"option.value\" [disabled]=\"option.disabled\">\n        @if (option.icon) {\n          <app-icon [fontIcon]=\"option.icon\"></app-icon>\n        }\n        <span>{{ option.label }}</span>\n      </mat-button-toggle>\n    }\n  </mat-button-toggle-group>\n} @else {\n  <div class=\"toggle-buttons\" [class.vertical]=\"vertical\">\n    @for (option of options; track option.value) {\n      <app-button\n        [text]=\"option.label\"\n        [fontIcon]=\"option.icon || ''\"\n        [color]=\"option.color || (isSelected(option) ? 'primary' : 'secondary')\"\n        [appearance]=\"isSelected(option) ? 'raised' : 'flat'\"\n        [disabled]=\"option.disabled || false\"\n        (btnClick)=\"onClick(option)\"\n        [class.selected]=\"isSelected(option)\"\n      ></app-button>\n    }\n  </div>\n}\n\n", styles: [".toggle-group{display:flex;flex-wrap:wrap;gap:8px}.toggle-buttons{display:flex;gap:8px;flex-wrap:wrap}.toggle-buttons.vertical{flex-direction:column}.toggle-buttons app-button.selected{opacity:1}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatButtonToggleModule }, { kind: "directive", type: i1.MatButtonToggleGroup, selector: "mat-button-toggle-group", inputs: ["appearance", "name", "vertical", "value", "multiple", "disabled", "hideSingleSelectionIndicator", "hideMultipleSelectionIndicator"], outputs: ["valueChange", "change"], exportAs: ["matButtonToggleGroup"] }, { kind: "component", type: i1.MatButtonToggle, selector: "mat-button-toggle", inputs: ["aria-label", "aria-labelledby", "id", "name", "value", "tabIndex", "disableRipple", "appearance", "checked", "disabled"], outputs: ["change"], exportAs: ["matButtonToggle"] }, { kind: "ngmodule", type: MatButtonModule }, { kind: "component", type: AppButtonComponent, selector: "app-button", inputs: ["color", "type", "disabled", "appearance", "class", "text", "fontIcon"], outputs: ["btnClick"] }, { kind: "component", type: IconComponent, selector: "app-icon", inputs: ["fontSet", "fontIcon", "iconType", "icomoonClass", "size", "isBlink", "fontBold", "color", "class", "counter", "isCovered"], outputs: ["iconClick"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ToggleButtonComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-toggle-button', standalone: true, imports: [CommonModule, MatButtonToggleModule, MatButtonModule, AppButtonComponent, IconComponent], template: "@if (multiple) {\n  <mat-button-toggle-group\n    [value]=\"selectedValues\"\n    [multiple]=\"true\"\n    [appearance]=\"appearance\"\n    [vertical]=\"vertical\"\n    (change)=\"onToggleChange($event.value)\"\n    class=\"toggle-group\"\n  >\n    @for (option of options; track option.value) {\n      <mat-button-toggle [value]=\"option.value\" [disabled]=\"option.disabled\">\n        @if (option.icon) {\n          <app-icon [fontIcon]=\"option.icon\"></app-icon>\n        }\n        <span>{{ option.label }}</span>\n      </mat-button-toggle>\n    }\n  </mat-button-toggle-group>\n} @else {\n  <div class=\"toggle-buttons\" [class.vertical]=\"vertical\">\n    @for (option of options; track option.value) {\n      <app-button\n        [text]=\"option.label\"\n        [fontIcon]=\"option.icon || ''\"\n        [color]=\"option.color || (isSelected(option) ? 'primary' : 'secondary')\"\n        [appearance]=\"isSelected(option) ? 'raised' : 'flat'\"\n        [disabled]=\"option.disabled || false\"\n        (btnClick)=\"onClick(option)\"\n        [class.selected]=\"isSelected(option)\"\n      ></app-button>\n    }\n  </div>\n}\n\n", styles: [".toggle-group{display:flex;flex-wrap:wrap;gap:8px}.toggle-buttons{display:flex;gap:8px;flex-wrap:wrap}.toggle-buttons.vertical{flex-direction:column}.toggle-buttons app-button.selected{opacity:1}\n"] }]
        }], propDecorators: { options: [{
                type: Input
            }], value: [{
                type: Input
            }], multiple: [{
                type: Input
            }], appearance: [{
                type: Input
            }], vertical: [{
                type: Input
            }], valueChange: [{
                type: Output
            }], buttonClick: [{
                type: Output
            }] } });
