// @lk/core/esm2022/lib/tools/button-checkbox/button-checkbox.component.mjs
import { CommonModule } from '@angular/common';
import { Component, forwardRef, Input } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { IconComponent } from '../app-icon/icon.component';
import { TooltipDirective } from '../../directives/tooltip.directive';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
let uniqueId = 0;
export class ButtonCheckboxComponent {
    constructor() {
        this.label = '';
        this.icon = '';
        this.id = `app-button-checkbox-${uniqueId++}`;
        this.disabled = false;
        this.value = false;
        this.onChange = () => { };
        this.onTouched = () => { };
    }
    onInputChange(event) {
        const target = event.target;
        this.value = target.checked;
        this.onChange(this.value);
        this.onTouched();
    }
    // ControlValueAccessor
    writeValue(value) {
        this.value = !!value;
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ButtonCheckboxComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: ButtonCheckboxComponent, isStandalone: true, selector: "app-button-checkbox", inputs: { label: "label", icon: "icon", id: "id", disabled: "disabled" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(() => ButtonCheckboxComponent),
                multi: true
            }
        ], ngImport: i0, template: "<div class=\"button-checkbox\" [appTooltip]=\"label\">\n  <input\n    type=\"checkbox\"\n    class=\"button-checkbox-wrapper\"\n    [id]=\"id\"\n    [checked]=\"value\"\n    [disabled]=\"disabled\"\n    (change)=\"onInputChange($event)\"\n  />\n  <label [for]=\"id\" class=\"button-checkbox-label\">\n    <app-icon *ngIf=\"icon\" [fontIcon]=\"icon\" [color]=\"value?'var(--primary-text-color)':''\" [size]=\"20\"></app-icon>\n    <span>{{ label }}</span>\n  </label>\n</div>\n", styles: [".button-checkbox{position:relative;display:block;max-width:100%}.button-checkbox-wrapper{position:absolute;opacity:0;cursor:pointer;height:0;width:0}.button-checkbox-wrapper:checked+.button-checkbox-label{background:var(--primary-color);color:#fff}.button-checkbox-wrapper:checked+.button-checkbox-label app-icon{color:#fff}.button-checkbox-label{display:flex;align-items:center;gap:.5em;padding:.5em .8em;background:#f8fafc;border:2px solid #e2e8f0;border-radius:.7em;cursor:pointer;transition:all .3s cubic-bezier(.4,0,.2,1);color:#1e293b;max-width:100%}.button-checkbox-label:hover{background:#f1f5f9;border-color:#cbd5e1;transform:translateY(-1px);box-shadow:0 4px 12px #0000001a}.button-checkbox-label app-icon{color:#3b82f6;transition:color .3s ease}.button-checkbox-label span{font-size:.9em;overflow:hidden;white-space:nowrap;text-overflow:ellipsis}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "component", type: IconComponent, selector: "app-icon", inputs: ["fontSet", "fontIcon", "iconType", "icomoonClass", "size", "isBlink", "fontBold", "color", "class", "counter", "isCovered"], outputs: ["iconClick"] }, { kind: "directive", type: TooltipDirective, selector: "[appTooltip]", inputs: ["appTooltip", "tooltipPosition", "tooltipShowDelay", "tooltipHideDelay", "tooltipDisabled"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ButtonCheckboxComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-button-checkbox', standalone: true, imports: [CommonModule, IconComponent, TooltipDirective], providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef(() => ButtonCheckboxComponent),
                            multi: true
                        }
                    ], template: "<div class=\"button-checkbox\" [appTooltip]=\"label\">\n  <input\n    type=\"checkbox\"\n    class=\"button-checkbox-wrapper\"\n    [id]=\"id\"\n    [checked]=\"value\"\n    [disabled]=\"disabled\"\n    (change)=\"onInputChange($event)\"\n  />\n  <label [for]=\"id\" class=\"button-checkbox-label\">\n    <app-icon *ngIf=\"icon\" [fontIcon]=\"icon\" [color]=\"value?'var(--primary-text-color)':''\" [size]=\"20\"></app-icon>\n    <span>{{ label }}</span>\n  </label>\n</div>\n", styles: [".button-checkbox{position:relative;display:block;max-width:100%}.button-checkbox-wrapper{position:absolute;opacity:0;cursor:pointer;height:0;width:0}.button-checkbox-wrapper:checked+.button-checkbox-label{background:var(--primary-color);color:#fff}.button-checkbox-wrapper:checked+.button-checkbox-label app-icon{color:#fff}.button-checkbox-label{display:flex;align-items:center;gap:.5em;padding:.5em .8em;background:#f8fafc;border:2px solid #e2e8f0;border-radius:.7em;cursor:pointer;transition:all .3s cubic-bezier(.4,0,.2,1);color:#1e293b;max-width:100%}.button-checkbox-label:hover{background:#f1f5f9;border-color:#cbd5e1;transform:translateY(-1px);box-shadow:0 4px 12px #0000001a}.button-checkbox-label app-icon{color:#3b82f6;transition:color .3s ease}.button-checkbox-label span{font-size:.9em;overflow:hidden;white-space:nowrap;text-overflow:ellipsis}\n"] }]
        }], propDecorators: { label: [{
                type: Input
            }], icon: [{
                type: Input
            }], id: [{
                type: Input
            }], disabled: [{
                type: Input
            }] } });
