// @lk/core/esm2022/lib/tools/app-selectbox/app-selectbox.component.mjs
import { Component, Input, Output, EventEmitter, forwardRef } from '@angular/core';
import { MatSelectModule } from '@angular/material/select';
import { FormsModule, NG_VALUE_ACCESSOR, FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatOptionModule } from '@angular/material/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/material/form-field";
import * as i3 from "@angular/material/select";
import * as i4 from "@angular/material/core";
export class AppSelectboxComponent {
    constructor() {
        this.options = [];
        this.displayExpr = '';
        this.valueField = '';
        this.placeholder = '';
        this.label = '';
        this.formGroup = new FormGroup({});
        this.formControlName = '';
        this.disabled = false;
        this.required = false;
        this.errorMessage = '';
        this.formControl = new FormControl('');
        this.control = new FormControl('');
        this.dataChange = new EventEmitter();
        // ControlValueAccessor implementation
        this.onChange = (value) => { };
        this.onTouched = () => { };
        this.formGroup.addControl(this.formControlName, this.formControl);
    }
    onSelectionChange(value) {
        this.data = value;
        this.dataChange.emit(this.data);
        this.onChange(value);
        this.onTouched();
    }
    getDisplay(option) {
        return this.displayExpr ? option?.[this.displayExpr] : option;
    }
    getValue(option) {
        return this.valueField ? option?.[this.valueField] : option;
    }
    // ControlValueAccessor methods
    writeValue(value) {
        this.data = value;
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: AppSelectboxComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: AppSelectboxComponent, isStandalone: true, selector: "app-selectbox", inputs: { options: "options", displayExpr: "displayExpr", valueField: "valueField", placeholder: "placeholder", label: "label", formGroup: "formGroup", formControlName: "formControlName", disabled: "disabled", required: "required", errorMessage: "errorMessage", data: "data" }, outputs: { dataChange: "dataChange" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(() => AppSelectboxComponent),
                multi: true
            }
        ], ngImport: i0, template: "<!-- <mat-form-field class=\"app-selectbox-field\" appearance=\"outline\">\n  \n  <mat-select\n    [formControl]=\"control\"\n    [placeholder]=\"placeholder\"\n    [disabled]=\"disabled\"\n    [value]=\"data\"\n    (selectionChange)=\"onSelectionChange($event.value)\"\n    [required]=\"required\"\n  >\n    <mat-option *ngFor=\"let option of options\" [value]=\"getValue(option)\">\n      {{ getDisplay(option) }}\n    </mat-option>\n  </mat-select>\n  <mat-error *ngIf=\"errorMessage\">{{ errorMessage }}</mat-error>\n</mat-form-field> -->\n\n\n<mat-form-field>\n  <mat-label *ngIf=\"label\">{{ label }}</mat-label>\n  <mat-select name=\"countryString\" [(value)]=\"data\" [placeholder]=\"placeholder\">\n    @for (option of options; track option.value) {\n      <mat-option [value]=\"getValue(option)\">{{ getDisplay(option) }}</mat-option>\n    }\n  </mat-select>\n</mat-form-field>", styles: [".app-selectbox-field{width:100%;margin-bottom:1rem}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "ngmodule", type: MatSelectModule }, { kind: "component", type: i2.MatFormField, selector: "mat-form-field", inputs: ["hideRequiredMarker", "color", "floatLabel", "appearance", "subscriptSizing", "hintLabel"], exportAs: ["matFormField"] }, { kind: "directive", type: i2.MatLabel, selector: "mat-label" }, { kind: "component", type: i3.MatSelect, selector: "mat-select", inputs: ["aria-describedby", "panelClass", "disabled", "disableRipple", "tabIndex", "hideSingleSelectionIndicator", "placeholder", "required", "multiple", "disableOptionCentering", "compareWith", "value", "aria-label", "aria-labelledby", "errorStateMatcher", "typeaheadDebounceInterval", "sortComparator", "id", "panelWidth"], outputs: ["openedChange", "opened", "closed", "selectionChange", "valueChange"], exportAs: ["matSelect"] }, { kind: "component", type: i4.MatOption, selector: "mat-option", inputs: ["value", "id", "disabled"], outputs: ["onSelectionChange"], exportAs: ["matOption"] }, { kind: "ngmodule", type: FormsModule }, { kind: "ngmodule", type: MatFormFieldModule }, { kind: "ngmodule", type: MatOptionModule }, { kind: "ngmodule", type: ReactiveFormsModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: AppSelectboxComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-selectbox', standalone: true, imports: [CommonModule, MatSelectModule, FormsModule, MatFormFieldModule, MatOptionModule, ReactiveFormsModule], providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef(() => AppSelectboxComponent),
                            multi: true
                        }
                    ], template: "<!-- <mat-form-field class=\"app-selectbox-field\" appearance=\"outline\">\n  \n  <mat-select\n    [formControl]=\"control\"\n    [placeholder]=\"placeholder\"\n    [disabled]=\"disabled\"\n    [value]=\"data\"\n    (selectionChange)=\"onSelectionChange($event.value)\"\n    [required]=\"required\"\n  >\n    <mat-option *ngFor=\"let option of options\" [value]=\"getValue(option)\">\n      {{ getDisplay(option) }}\n    </mat-option>\n  </mat-select>\n  <mat-error *ngIf=\"errorMessage\">{{ errorMessage }}</mat-error>\n</mat-form-field> -->\n\n\n<mat-form-field>\n  <mat-label *ngIf=\"label\">{{ label }}</mat-label>\n  <mat-select name=\"countryString\" [(value)]=\"data\" [placeholder]=\"placeholder\">\n    @for (option of options; track option.value) {\n      <mat-option [value]=\"getValue(option)\">{{ getDisplay(option) }}</mat-option>\n    }\n  </mat-select>\n</mat-form-field>", styles: [".app-selectbox-field{width:100%;margin-bottom:1rem}\n"] }]
        }], ctorParameters: () => [], propDecorators: { options: [{
                type: Input
            }], displayExpr: [{
                type: Input
            }], valueField: [{
                type: Input
            }], placeholder: [{
                type: Input
            }], label: [{
                type: Input
            }], formGroup: [{
                type: Input
            }], formControlName: [{
                type: Input
            }], disabled: [{
                type: Input
            }], required: [{
                type: Input
            }], errorMessage: [{
                type: Input
            }], data: [{
                type: Input
            }], dataChange: [{
                type: Output
            }] } });
