// @lk/template/esm2022/lib/template.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import * as i0 from "@angular/core";
// Import all template components here
// This module will be imported by the main app
export class TemplateModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TemplateModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: TemplateModule, imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            RouterModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TemplateModule, imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            RouterModule] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TemplateModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                    // Components will be declared here
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        RouterModule
                    ],
                    exports: [
                    // Export components that should be available to consuming apps
                    ]
                }]
        }] });
