// @lk/core/esm2022/lib/tools/json-reflect/json-reflect.component.mjs
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatInputModule } from '@angular/material/input';
import { MatFormFieldModule } from '@angular/material/form-field';
import { AppButtonComponent } from '../app-button/app-button.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
export class JsonReflectComponent {
    constructor() {
        this.json = {};
        this.editable = true;
        this.showControls = true;
        this.height = '400px';
        this.jsonChange = new EventEmitter();
        this.save = new EventEmitter();
        this.cancel = new EventEmitter();
        this.jsonString = '';
        this.isEditing = false;
        this.errorMessage = '';
    }
    ngOnInit() {
        this.updateJsonString();
    }
    ngOnChanges(changes) {
        if (changes['json'] && !this.isEditing) {
            this.updateJsonString();
        }
    }
    updateJsonString() {
        try {
            this.jsonString = JSON.stringify(this.json, null, 2);
            this.errorMessage = '';
        }
        catch (error) {
            this.jsonString = '';
            this.errorMessage = 'Invalid JSON';
        }
    }
    toggleEdit() {
        if (this.editable) {
            this.isEditing = !this.isEditing;
            if (!this.isEditing) {
                this.updateJsonString();
            }
        }
    }
    onJsonChange() {
        try {
            const parsed = JSON.parse(this.jsonString);
            this.json = parsed;
            this.errorMessage = '';
            this.jsonChange.emit(parsed);
        }
        catch (error) {
            this.errorMessage = 'Invalid JSON format';
        }
    }
    onSave() {
        try {
            const parsed = JSON.parse(this.jsonString);
            this.json = parsed;
            this.errorMessage = '';
            this.isEditing = false;
            this.save.emit(parsed);
        }
        catch (error) {
            this.errorMessage = 'Invalid JSON format. Cannot save.';
        }
    }
    onCancel() {
        this.updateJsonString();
        this.isEditing = false;
        this.errorMessage = '';
        this.cancel.emit();
    }
    formatJson() {
        try {
            const parsed = JSON.parse(this.jsonString);
            this.jsonString = JSON.stringify(parsed, null, 2);
            this.errorMessage = '';
        }
        catch (error) {
            this.errorMessage = 'Invalid JSON format';
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: JsonReflectComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: JsonReflectComponent, isStandalone: true, selector: "app-json-reflect", inputs: { json: "json", editable: "editable", showControls: "showControls", height: "height" }, outputs: { jsonChange: "jsonChange", save: "save", cancel: "cancel" }, usesOnChanges: true, ngImport: i0, template: "<div class=\"json-reflect-container\" [style.height]=\"height\">\n  @if (showControls && editable) {\n    <div class=\"json-controls\">\n      @if (!isEditing) {\n        <app-button\n          text=\"Edit\"\n          fontIcon=\"edit\"\n          appearance=\"raised\"\n          color=\"primary\"\n          (btnClick)=\"toggleEdit()\"\n        ></app-button>\n      } @else {\n        <app-button\n          text=\"Save\"\n          fontIcon=\"save\"\n          appearance=\"raised\"\n          color=\"primary\"\n          (btnClick)=\"onSave()\"\n        ></app-button>\n        <app-button\n          text=\"Cancel\"\n          fontIcon=\"cancel\"\n          appearance=\"flat\"\n          color=\"warn\"\n          (btnClick)=\"onCancel()\"\n        ></app-button>\n        <app-button\n          text=\"Format\"\n          fontIcon=\"code\"\n          appearance=\"flat\"\n          (btnClick)=\"formatJson()\"\n        ></app-button>\n      }\n    </div>\n  }\n\n  @if (errorMessage) {\n    <div class=\"error-message\">{{ errorMessage }}</div>\n  }\n\n  @if (isEditing && editable) {\n    <textarea\n      class=\"json-editor\"\n      [(ngModel)]=\"jsonString\"\n      (input)=\"onJsonChange()\"\n      placeholder=\"Enter JSON here...\"\n    ></textarea>\n  } @else {\n    <pre class=\"json-viewer\">{{ jsonString }}</pre>\n  }\n</div>\n\n", styles: [".json-reflect-container{display:flex;flex-direction:column;border:1px solid #e0e0e0;border-radius:4px;overflow:hidden}.json-controls{display:flex;gap:8px;padding:8px;background-color:#f5f5f5;border-bottom:1px solid #e0e0e0}.error-message{padding:8px 16px;background-color:#ffebee;color:#c62828;font-size:14px;border-bottom:1px solid #e0e0e0}.json-editor{flex:1;padding:16px;font-family:Courier New,monospace;font-size:14px;line-height:1.5;border:none;outline:none;resize:none;background-color:#fafafa}.json-viewer{flex:1;padding:16px;margin:0;font-family:Courier New,monospace;font-size:14px;line-height:1.5;overflow:auto;background-color:#fafafa;white-space:pre-wrap;word-wrap:break-word}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: FormsModule }, { kind: "directive", type: i1.DefaultValueAccessor, selector: "input:not([type=checkbox])[formControlName],textarea[formControlName],input:not([type=checkbox])[formControl],textarea[formControl],input:not([type=checkbox])[ngModel],textarea[ngModel],[ngDefaultControl]" }, { kind: "directive", type: i1.NgControlStatus, selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i1.NgModel, selector: "[ngModel]:not([formControlName]):not([formControl])", inputs: ["name", "disabled", "ngModel", "ngModelOptions"], outputs: ["ngModelChange"], exportAs: ["ngModel"] }, { kind: "ngmodule", type: MatButtonModule }, { kind: "ngmodule", type: MatIconModule }, { kind: "ngmodule", type: MatInputModule }, { kind: "ngmodule", type: MatFormFieldModule }, { kind: "component", type: AppButtonComponent, selector: "app-button", inputs: ["color", "type", "disabled", "appearance", "class", "text", "fontIcon"], outputs: ["btnClick"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: JsonReflectComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-json-reflect', standalone: true, imports: [
                        CommonModule,
                        FormsModule,
                        MatButtonModule,
                        MatIconModule,
                        MatInputModule,
                        MatFormFieldModule,
                        AppButtonComponent
                    ], template: "<div class=\"json-reflect-container\" [style.height]=\"height\">\n  @if (showControls && editable) {\n    <div class=\"json-controls\">\n      @if (!isEditing) {\n        <app-button\n          text=\"Edit\"\n          fontIcon=\"edit\"\n          appearance=\"raised\"\n          color=\"primary\"\n          (btnClick)=\"toggleEdit()\"\n        ></app-button>\n      } @else {\n        <app-button\n          text=\"Save\"\n          fontIcon=\"save\"\n          appearance=\"raised\"\n          color=\"primary\"\n          (btnClick)=\"onSave()\"\n        ></app-button>\n        <app-button\n          text=\"Cancel\"\n          fontIcon=\"cancel\"\n          appearance=\"flat\"\n          color=\"warn\"\n          (btnClick)=\"onCancel()\"\n        ></app-button>\n        <app-button\n          text=\"Format\"\n          fontIcon=\"code\"\n          appearance=\"flat\"\n          (btnClick)=\"formatJson()\"\n        ></app-button>\n      }\n    </div>\n  }\n\n  @if (errorMessage) {\n    <div class=\"error-message\">{{ errorMessage }}</div>\n  }\n\n  @if (isEditing && editable) {\n    <textarea\n      class=\"json-editor\"\n      [(ngModel)]=\"jsonString\"\n      (input)=\"onJsonChange()\"\n      placeholder=\"Enter JSON here...\"\n    ></textarea>\n  } @else {\n    <pre class=\"json-viewer\">{{ jsonString }}</pre>\n  }\n</div>\n\n", styles: [".json-reflect-container{display:flex;flex-direction:column;border:1px solid #e0e0e0;border-radius:4px;overflow:hidden}.json-controls{display:flex;gap:8px;padding:8px;background-color:#f5f5f5;border-bottom:1px solid #e0e0e0}.error-message{padding:8px 16px;background-color:#ffebee;color:#c62828;font-size:14px;border-bottom:1px solid #e0e0e0}.json-editor{flex:1;padding:16px;font-family:Courier New,monospace;font-size:14px;line-height:1.5;border:none;outline:none;resize:none;background-color:#fafafa}.json-viewer{flex:1;padding:16px;margin:0;font-family:Courier New,monospace;font-size:14px;line-height:1.5;overflow:auto;background-color:#fafafa;white-space:pre-wrap;word-wrap:break-word}\n"] }]
        }], propDecorators: { json: [{
                type: Input
            }], editable: [{
                type: Input
            }], showControls: [{
                type: Input
            }], height: [{
                type: Input
            }], jsonChange: [{
                type: Output
            }], save: [{
                type: Output
            }], cancel: [{
                type: Output
            }] } });
