// @lk/core/esm2022/lib/tools/pdf-viewer/pdf-viewer.component.mjs
import { Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { AppButtonComponent } from '../app-button/app-button.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
export class PdfViewerComponent {
    constructor(sanitizer) {
        this.sanitizer = sanitizer;
        this.src = null;
        this.height = '600px';
        this.width = '100%';
        this.showToolbar = true;
        this.downloadFileName = 'document.pdf';
        this.safeSrc = null;
        this.pdfUrl = '';
    }
    ngOnInit() {
        this.updatePdfSource();
    }
    ngOnChanges(changes) {
        if (changes['src']) {
            this.updatePdfSource();
        }
    }
    updatePdfSource() {
        if (!this.src) {
            this.safeSrc = null;
            this.pdfUrl = '';
            return;
        }
        if (typeof this.src === 'string') {
            this.pdfUrl = this.src;
            this.safeSrc = this.sanitizer.bypassSecurityTrustResourceUrl(this.src);
        }
        else if (this.src instanceof ArrayBuffer) {
            const blob = new Blob([this.src], { type: 'application/pdf' });
            const url = URL.createObjectURL(blob);
            this.pdfUrl = url;
            this.safeSrc = this.sanitizer.bypassSecurityTrustResourceUrl(url);
        }
    }
    downloadPdf() {
        if (this.pdfUrl) {
            const link = document.createElement('a');
            link.href = this.pdfUrl;
            link.download = this.downloadFileName;
            link.click();
        }
    }
    printPdf() {
        if (this.pdfUrl) {
            const printWindow = window.open(this.pdfUrl, '_blank');
            if (printWindow) {
                printWindow.onload = () => {
                    printWindow.print();
                };
            }
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: PdfViewerComponent, deps: [{ token: i1.DomSanitizer }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: PdfViewerComponent, isStandalone: true, selector: "app-pdf-viewer", inputs: { src: "src", height: "height", width: "width", showToolbar: "showToolbar", downloadFileName: "downloadFileName" }, usesOnChanges: true, ngImport: i0, template: "<div class=\"pdf-viewer-container\" [style.width]=\"width\" [style.height]=\"height\">\n  @if (showToolbar) {\n    <div class=\"pdf-toolbar\">\n      <app-button\n        text=\"Download\"\n        fontIcon=\"download\"\n        appearance=\"flat\"\n        color=\"primary\"\n        (btnClick)=\"downloadPdf()\"\n      ></app-button>\n      <app-button\n        text=\"Print\"\n        fontIcon=\"print\"\n        appearance=\"flat\"\n        color=\"primary\"\n        (btnClick)=\"printPdf()\"\n      ></app-button>\n    </div>\n  }\n\n  @if (safeSrc) {\n    <iframe\n      [src]=\"safeSrc\"\n      class=\"pdf-iframe\"\n      [style.height]=\"showToolbar ? 'calc(' + height + ' - 48px)' : height\"\n    ></iframe>\n  } @else {\n    <div class=\"pdf-placeholder\">\n      <p>No PDF to display</p>\n    </div>\n  }\n</div>\n\n", styles: [".pdf-viewer-container{display:flex;flex-direction:column;border:1px solid #e0e0e0;border-radius:4px;overflow:hidden}.pdf-toolbar{display:flex;gap:8px;padding:8px;background-color:#f5f5f5;border-bottom:1px solid #e0e0e0}.pdf-iframe{width:100%;border:none;flex:1}.pdf-placeholder{display:flex;align-items:center;justify-content:center;height:100%;color:#9e9e9e;font-size:16px}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatButtonModule }, { kind: "ngmodule", type: MatIconModule }, { kind: "component", type: AppButtonComponent, selector: "app-button", inputs: ["color", "type", "disabled", "appearance", "class", "text", "fontIcon"], outputs: ["btnClick"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: PdfViewerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-pdf-viewer', standalone: true, imports: [CommonModule, MatButtonModule, MatIconModule, AppButtonComponent], template: "<div class=\"pdf-viewer-container\" [style.width]=\"width\" [style.height]=\"height\">\n  @if (showToolbar) {\n    <div class=\"pdf-toolbar\">\n      <app-button\n        text=\"Download\"\n        fontIcon=\"download\"\n        appearance=\"flat\"\n        color=\"primary\"\n        (btnClick)=\"downloadPdf()\"\n      ></app-button>\n      <app-button\n        text=\"Print\"\n        fontIcon=\"print\"\n        appearance=\"flat\"\n        color=\"primary\"\n        (btnClick)=\"printPdf()\"\n      ></app-button>\n    </div>\n  }\n\n  @if (safeSrc) {\n    <iframe\n      [src]=\"safeSrc\"\n      class=\"pdf-iframe\"\n      [style.height]=\"showToolbar ? 'calc(' + height + ' - 48px)' : height\"\n    ></iframe>\n  } @else {\n    <div class=\"pdf-placeholder\">\n      <p>No PDF to display</p>\n    </div>\n  }\n</div>\n\n", styles: [".pdf-viewer-container{display:flex;flex-direction:column;border:1px solid #e0e0e0;border-radius:4px;overflow:hidden}.pdf-toolbar{display:flex;gap:8px;padding:8px;background-color:#f5f5f5;border-bottom:1px solid #e0e0e0}.pdf-iframe{width:100%;border:none;flex:1}.pdf-placeholder{display:flex;align-items:center;justify-content:center;height:100%;color:#9e9e9e;font-size:16px}\n"] }]
        }], ctorParameters: () => [{ type: i1.DomSanitizer }], propDecorators: { src: [{
                type: Input
            }], height: [{
                type: Input
            }], width: [{
                type: Input
            }], showToolbar: [{
                type: Input
            }], downloadFileName: [{
                type: Input
            }] } });
