// @lk/core/esm2022/lib/core-event.service.mjs
import { Injectable, signal, computed, effect, runInInjectionContext } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Core Event Service using Angular Signals for cross-library communication
 * This service provides a centralized event bus for the application
 */
export class CoreEventService {
    constructor(injector) {
        this.injector = injector;
        // Private signals for internal state management
        this._menu = signal(null);
        this._authState = signal(null);
        this._breadcrumb = signal(null);
        this._notification = signal(null);
        // Public read-only computed signals
        this.menu = computed(() => this._menu());
        this.authState = computed(() => this._authState());
        this.breadcrumb = computed(() => this._breadcrumb());
        this.notification = computed(() => this._notification());
    }
    // Menu updaters
    setMenu(menu) {
        this._menu.set(menu);
    }
    clearMenu() {
        this._menu.set(null);
    }
    // Auth state updaters
    setAuth(state) {
        this._authState.set(state);
    }
    clearAuth() {
        this._authState.set(null);
    }
    // Breadcrumb updaters
    setBreadcrumb(breadcrumb) {
        if (breadcrumb === null) {
            this._breadcrumb.set(null);
        }
        else if (Array.isArray(breadcrumb)) {
            this._breadcrumb.set(breadcrumb);
        }
        else {
            // Single item, convert to array
            this._breadcrumb.set([breadcrumb]);
        }
    }
    clearBreadcrumb() {
        this._breadcrumb.set(null);
    }
    // Notification updaters
    setNotification(notification) {
        this._notification.set(notification);
    }
    clearNotification() {
        this._notification.set(null);
    }
    // Convenience methods: run callback on signal changes (returns a destroy function)
    onMenuChange(cb) {
        return runInInjectionContext(this.injector, () => {
            const runner = effect(() => {
                cb(this._menu());
            });
            return () => runner.destroy(); // destroy the effect to cleanup
        });
    }
    onAuthChange(cb) {
        return runInInjectionContext(this.injector, () => {
            const runner = effect(() => {
                cb(this._authState());
            });
            return () => runner.destroy();
        });
    }
    onBreadcrumbChange(cb) {
        return runInInjectionContext(this.injector, () => {
            const runner = effect(() => {
                cb(this._breadcrumb());
            });
            return () => runner.destroy();
        });
    }
    onNotificationChange(cb) {
        return runInInjectionContext(this.injector, () => {
            const runner = effect(() => {
                cb(this._notification());
            });
            return () => runner.destroy();
        });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: CoreEventService, deps: [{ token: i0.Injector }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: CoreEventService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: CoreEventService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: () => [{ type: i0.Injector }] });
