// @lk/core/esm2022/lib/services/api-config.service.mjs
import { Injectable, InjectionToken, Inject, Optional } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * Injection token for API base URL configuration
 */
export const API_BASE_URL = new InjectionToken('API_BASE_URL');
/**
 * Service to provide API configuration
 * The consuming application should provide API_BASE_URL token
 */
export class ApiConfigService {
    constructor(apiUrl) {
        // Default to empty string if not provided - consuming app must provide
        this.apiUrl = apiUrl || '';
    }
    getApiUrl() {
        if (!this.apiUrl) {
            console.warn('API_BASE_URL not provided. Please provide API_BASE_URL token in your app configuration.');
        }
        return this.apiUrl;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ApiConfigService, deps: [{ token: API_BASE_URL, optional: true }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ApiConfigService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ApiConfigService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: undefined, decorators: [{
                    type: Optional
                }, {
                    type: Inject,
                    args: [API_BASE_URL]
                }] }] });
