// @lk/core/esm2022/lib/services/chart.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class ChartService {
    constructor() {
        this.defaultOptions = {
            chart: {
                type: 'line'
            },
            title: {
                text: 'Chart'
            },
            credits: {
                enabled: false
            },
            legend: {
                enabled: true
            },
            tooltip: {
                enabled: true
            }
        };
    }
    /**
     * Create a chart instance
     * @param container - The DOM element or element ID to render the chart in
     * @param options - Chart configuration options
     * @returns Highcharts chart instance
     */
    createChart(container, options) {
        if (typeof Highcharts === 'undefined') {
            console.error('Highcharts library is not loaded. Please include Highcharts in your project.');
            return null;
        }
        const mergedOptions = this.mergeOptions(options);
        return Highcharts.chart(container, mergedOptions);
    }
    /**
     * Update an existing chart
     * @param chartInstance - The Highcharts chart instance
     * @param options - Chart configuration options to update
     */
    updateChart(chartInstance, options) {
        if (!chartInstance) {
            console.error('Chart instance is not provided');
            return;
        }
        const mergedOptions = this.mergeOptions(options);
        chartInstance.update(mergedOptions, true);
    }
    /**
     * Destroy a chart instance
     * @param chartInstance - The Highcharts chart instance to destroy
     */
    destroyChart(chartInstance) {
        if (chartInstance && typeof chartInstance.destroy === 'function') {
            chartInstance.destroy();
        }
    }
    /**
     * Get default chart options
     * @param type - Chart type
     * @returns Default chart options for the specified type
     */
    getDefaultOptions(type = 'line') {
        return {
            ...this.defaultOptions,
            chart: {
                ...this.defaultOptions.chart,
                type: type
            }
        };
    }
    /**
     * Create chart options object with provided input
     * @param input - Partial chart options
     * @returns Complete chart options object
     */
    createChartOptions(input) {
        return this.mergeOptions(input);
    }
    mergeOptions(options) {
        const merged = {
            ...this.defaultOptions,
            ...options
        };
        // Deep merge chart, title, and other nested objects
        if (options.chart) {
            merged.chart = {
                ...this.defaultOptions.chart,
                ...options.chart
            };
        }
        if (options.title) {
            merged.title = {
                ...this.defaultOptions.title,
                ...options.title
            };
        }
        return merged;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ChartService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ChartService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ChartService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }] });
