// @lk/core/esm2022/lib/services/http-calls.service.mjs
import { Injectable } from '@angular/core';
import { HttpHeaders, HttpParams } from '@angular/common/http';
import { throwError } from 'rxjs';
import { catchError, map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
export class HttpCallsService {
    constructor(http) {
        this.http = http;
        this.handleError = (error) => {
            const errorResponse = {
                message: error.message || 'An error occurred',
                status: error.status,
                statusText: error.statusText,
                error: error.error
            };
            return throwError(() => errorResponse);
        };
    }
    /**
     * Send GET request
     * @param url - The URL to send the GET request to
     * @param options - Optional HTTP options
     * @returns Observable with the response data
     */
    sendGet(url, options) {
        const httpOptions = this.prepareOptions(options);
        // Ensure observe is 'body' for simple methods unless explicitly set
        if (!options?.observe) {
            httpOptions.observe = 'body';
        }
        return this.http.get(url, httpOptions).pipe(catchError(this.handleError));
    }
    /**
     * Send POST request
     * @param url - The URL to send the POST request to
     * @param body - The request body
     * @param options - Optional HTTP options
     * @returns Observable with the response data
     */
    sendPost(url, body, options) {
        const httpOptions = this.prepareOptions(options);
        // Ensure observe is 'body' for simple methods unless explicitly set
        if (!options?.observe) {
            httpOptions.observe = 'body';
        }
        return this.http.post(url, body, httpOptions).pipe(catchError(this.handleError));
    }
    /**
     * Send PUT request
     * @param url - The URL to send the PUT request to
     * @param body - The request body
     * @param options - Optional HTTP options
     * @returns Observable with the response data
     */
    sendPut(url, body, options) {
        const httpOptions = this.prepareOptions(options);
        // Ensure observe is 'body' for simple methods unless explicitly set
        if (!options?.observe) {
            httpOptions.observe = 'body';
        }
        return this.http.put(url, body, httpOptions).pipe(catchError(this.handleError));
    }
    /**
     * Send PATCH request
     * @param url - The URL to send the PATCH request to
     * @param body - The request body
     * @param options - Optional HTTP options
     * @returns Observable with the response data
     */
    sendPatch(url, body, options) {
        const httpOptions = this.prepareOptions(options);
        // Ensure observe is 'body' for simple methods unless explicitly set
        if (!options?.observe) {
            httpOptions.observe = 'body';
        }
        return this.http.patch(url, body, httpOptions).pipe(catchError(this.handleError));
    }
    /**
     * Send DELETE request
     * @param url - The URL to send the DELETE request to
     * @param options - Optional HTTP options
     * @returns Observable with the response data
     */
    sendDelete(url, options) {
        const httpOptions = this.prepareOptions(options);
        // Ensure observe is 'body' for simple methods unless explicitly set
        if (!options?.observe) {
            httpOptions.observe = 'body';
        }
        return this.http.delete(url, httpOptions).pipe(catchError(this.handleError));
    }
    /**
     * Get full HTTP response (including headers, status, etc.)
     * @param url - The URL to send the GET request to
     * @param options - Optional HTTP options
     * @returns Observable with the full HTTP response
     */
    sendGetResponse(url, options) {
        const baseOptions = this.prepareOptions(options);
        const httpOptions = {
            ...baseOptions,
            observe: 'response'
        };
        return this.http.get(url, httpOptions).pipe(map((response) => ({
            data: response.body,
            status: response.status,
            statusText: response.statusText,
            headers: response.headers
        })), catchError(this.handleError));
    }
    /**
     * Post and get full HTTP response
     * @param url - The URL to send the POST request to
     * @param body - The request body
     * @param options - Optional HTTP options
     * @returns Observable with the full HTTP response
     */
    sendPostResponse(url, body, options) {
        const baseOptions = this.prepareOptions(options);
        const httpOptions = {
            ...baseOptions,
            observe: 'response'
        };
        return this.http.post(url, body, httpOptions).pipe(map((response) => ({
            data: response.body,
            status: response.status,
            statusText: response.statusText,
            headers: response.headers
        })), catchError(this.handleError));
    }
    prepareOptions(options) {
        const defaultOptions = {
            headers: new HttpHeaders({
                'Content-Type': 'application/json'
            }),
            observe: 'body',
            responseType: 'json'
        };
        if (options) {
            if (options.headers) {
                defaultOptions.headers = options.headers instanceof HttpHeaders
                    ? options.headers
                    : new HttpHeaders(options.headers);
            }
            if (options.params) {
                defaultOptions.params = options.params instanceof HttpParams
                    ? options.params
                    : new HttpParams({ fromObject: options.params });
            }
            if (options.observe) {
                defaultOptions.observe = options.observe;
            }
            if (options.reportProgress !== undefined) {
                defaultOptions.reportProgress = options.reportProgress;
            }
            if (options.responseType) {
                defaultOptions.responseType = options.responseType;
            }
            if (options.withCredentials !== undefined) {
                defaultOptions.withCredentials = options.withCredentials;
            }
        }
        return defaultOptions;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: HttpCallsService, deps: [{ token: i1.HttpClient }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: HttpCallsService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: HttpCallsService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.HttpClient }] });
