// @lk/core/esm2022/lib/services/snackbar.service.mjs
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/snack-bar";
export class SnackbarService {
    constructor(snackBar) {
        this.snackBar = snackBar;
        this.defaultConfig = {
            duration: 3000,
            horizontalPosition: 'end',
            verticalPosition: 'bottom',
            panelClass: []
        };
    }
    /**
     * Show success message
     * @param message - Message to display
     * @param config - Optional snackbar configuration
     * @returns Snackbar reference
     */
    success(message, config) {
        return this.show(message, {
            ...config,
            panelClass: ['success-snackbar', ...(config?.panelClass ? (Array.isArray(config.panelClass) ? config.panelClass : [config.panelClass]) : [])]
        });
    }
    /**
     * Show error message
     * @param message - Message to display
     * @param config - Optional snackbar configuration
     * @returns Snackbar reference
     */
    error(message, config) {
        return this.show(message, {
            ...config,
            panelClass: ['error-snackbar', ...(config?.panelClass ? (Array.isArray(config.panelClass) ? config.panelClass : [config.panelClass]) : [])]
        });
    }
    /**
     * Show warning message
     * @param message - Message to display
     * @param config - Optional snackbar configuration
     * @returns Snackbar reference
     */
    warning(message, config) {
        return this.show(message, {
            ...config,
            panelClass: ['warning-snackbar', ...(config?.panelClass ? (Array.isArray(config.panelClass) ? config.panelClass : [config.panelClass]) : [])]
        });
    }
    /**
     * Show info message
     * @param message - Message to display
     * @param config - Optional snackbar configuration
     * @returns Snackbar reference
     */
    info(message, config) {
        return this.show(message, {
            ...config,
            panelClass: ['info-snackbar', ...(config?.panelClass ? (Array.isArray(config.panelClass) ? config.panelClass : [config.panelClass]) : [])]
        });
    }
    /**
     * Show custom message
     * @param message - Message to display
     * @param config - Optional snackbar configuration
     * @returns Snackbar reference
     */
    show(message, config) {
        const snackbarConfig = {
            ...this.defaultConfig,
            duration: config?.duration ?? this.defaultConfig.duration,
            horizontalPosition: config?.horizontalPosition ?? this.defaultConfig.horizontalPosition,
            verticalPosition: config?.verticalPosition ?? this.defaultConfig.verticalPosition,
            panelClass: config?.panelClass ?? this.defaultConfig.panelClass
        };
        const snackbarRef = this.snackBar.open(message, config?.action || 'Close', snackbarConfig);
        if (config?.actionCallback) {
            snackbarRef.onAction().subscribe(() => {
                config.actionCallback();
            });
        }
        return snackbarRef;
    }
    /**
     * Dismiss the currently open snackbar
     */
    dismiss() {
        this.snackBar.dismiss();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: SnackbarService, deps: [{ token: i1.MatSnackBar }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: SnackbarService, providedIn: 'root' }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: SnackbarService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: () => [{ type: i1.MatSnackBar }] });
