// @lk/core/esm2022/lib/tools/tabs/tabs.component.mjs
import { Component, Input, ContentChildren, ViewChild } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatTabsModule } from '@angular/material/tabs';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/material/tabs";
export class TabComponent {
    constructor() {
        this.label = '';
        this.disabled = false;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TabComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: TabComponent, isStandalone: true, selector: "app-tab", inputs: { label: "label", disabled: "disabled" }, viewQueries: [{ propertyName: "content", first: true, predicate: ["tabContent"], descendants: true, static: true }], ngImport: i0, template: `
    <ng-template #tabContent>
      <ng-content></ng-content>
    </ng-template>
  `, isInline: true, dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatTabsModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TabComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-tab',
                    standalone: true,
                    imports: [CommonModule, MatTabsModule],
                    template: `
    <ng-template #tabContent>
      <ng-content></ng-content>
    </ng-template>
  `
                }]
        }], propDecorators: { label: [{
                type: Input
            }], disabled: [{
                type: Input
            }], content: [{
                type: ViewChild,
                args: ['tabContent', { static: true }]
            }] } });
export class TabsComponent {
    constructor() {
        this.selectedIndex = 0;
        this.animationDuration = '500ms';
        this.dynamicHeight = false;
        this.color = 'primary';
    }
    ngAfterContentInit() {
        // Tabs are managed by content projection
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TabsComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: TabsComponent, isStandalone: true, selector: "app-tabs", inputs: { selectedIndex: "selectedIndex", animationDuration: "animationDuration", dynamicHeight: "dynamicHeight", backgroundColor: "backgroundColor", color: "color" }, queries: [{ propertyName: "tabs", predicate: TabComponent }], ngImport: i0, template: "<mat-tab-group\n  [selectedIndex]=\"selectedIndex\"\n  [animationDuration]=\"animationDuration\"\n  [dynamicHeight]=\"dynamicHeight\"\n  [color]=\"color\"\n  class=\"app-tabs\"\n  [style.backgroundColor]=\"backgroundColor\"\n>\n  <mat-tab *ngFor=\"let tab of tabs\" [label]=\"tab.label\" [disabled]=\"tab.disabled\">\n    <ng-container *ngTemplateOutlet=\"tab.content\"></ng-container>\n  </mat-tab>\n</mat-tab-group>\n\n", styles: [".app-tabs{width:100%;height:100%}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i1.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet", "ngTemplateOutletInjector"] }, { kind: "ngmodule", type: MatTabsModule }, { kind: "component", type: i2.MatTab, selector: "mat-tab", inputs: ["disabled", "label", "aria-label", "aria-labelledby", "labelClass", "bodyClass"], exportAs: ["matTab"] }, { kind: "component", type: i2.MatTabGroup, selector: "mat-tab-group", inputs: ["color", "fitInkBarToContent", "mat-stretch-tabs", "dynamicHeight", "selectedIndex", "headerPosition", "animationDuration", "contentTabIndex", "disablePagination", "disableRipple", "preserveContent", "backgroundColor"], outputs: ["selectedIndexChange", "focusChange", "animationDone", "selectedTabChange"], exportAs: ["matTabGroup"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TabsComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-tabs', standalone: true, imports: [CommonModule, MatTabsModule, TabComponent], template: "<mat-tab-group\n  [selectedIndex]=\"selectedIndex\"\n  [animationDuration]=\"animationDuration\"\n  [dynamicHeight]=\"dynamicHeight\"\n  [color]=\"color\"\n  class=\"app-tabs\"\n  [style.backgroundColor]=\"backgroundColor\"\n>\n  <mat-tab *ngFor=\"let tab of tabs\" [label]=\"tab.label\" [disabled]=\"tab.disabled\">\n    <ng-container *ngTemplateOutlet=\"tab.content\"></ng-container>\n  </mat-tab>\n</mat-tab-group>\n\n", styles: [".app-tabs{width:100%;height:100%}\n"] }]
        }], propDecorators: { selectedIndex: [{
                type: Input
            }], animationDuration: [{
                type: Input
            }], dynamicHeight: [{
                type: Input
            }], backgroundColor: [{
                type: Input
            }], color: [{
                type: Input
            }], tabs: [{
                type: ContentChildren,
                args: [TabComponent]
            }] } });
