// @lk/core/esm2022/lib/tools/chart/chart.component.mjs
import { Component, Input, ViewChild } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
import * as i1 from "../../services/chart.service";
export class ChartComponent {
    constructor(chartService) {
        this.chartService = chartService;
        this.height = '100%';
        this.width = '100%';
        this.chartInstance = null;
    }
    ngOnInit() {
        if (this.chartOptions$) {
            this.subscription = this.chartOptions$.subscribe(options => {
                this.renderChart(options);
            });
        }
        else if (this.chartOptions) {
            this.renderChart(this.chartOptions);
        }
    }
    ngOnChanges(changes) {
        if (changes['chartOptions'] && this.chartOptions && !this.chartOptions$) {
            this.renderChart(this.chartOptions);
        }
    }
    ngOnDestroy() {
        if (this.subscription) {
            this.subscription.unsubscribe();
        }
        if (this.chartInstance) {
            this.chartInstance.destroy();
        }
    }
    renderChart(options) {
        if (this.chartContainer) {
            if (this.chartInstance) {
                this.chartInstance.destroy();
            }
            this.chartInstance = this.chartService.createChart(this.chartContainer.nativeElement, options);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ChartComponent, deps: [{ token: i1.ChartService }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: ChartComponent, isStandalone: true, selector: "app-chart", inputs: { chartOptions$: "chartOptions$", chartOptions: "chartOptions", height: "height", width: "width" }, viewQueries: [{ propertyName: "chartContainer", first: true, predicate: ["chartContainer"], descendants: true, static: true }], usesOnChanges: true, ngImport: i0, template: "<div class=\"chart-container\" [style.width]=\"width\" [style.height]=\"height\">\n  <div #chartContainer class=\"chart-content\"></div>\n</div>\n\n", styles: [".chart-container{width:100%;border:1px solid #e0e0e0;border-radius:4px;padding:16px;background-color:#fff}.chart-content{width:100%;height:100%}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ChartComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-chart', standalone: true, imports: [CommonModule], template: "<div class=\"chart-container\" [style.width]=\"width\" [style.height]=\"height\">\n  <div #chartContainer class=\"chart-content\"></div>\n</div>\n\n", styles: [".chart-container{width:100%;border:1px solid #e0e0e0;border-radius:4px;padding:16px;background-color:#fff}.chart-content{width:100%;height:100%}\n"] }]
        }], ctorParameters: () => [{ type: i1.ChartService }], propDecorators: { chartOptions$: [{
                type: Input
            }], chartOptions: [{
                type: Input
            }], height: [{
                type: Input
            }], width: [{
                type: Input
            }], chartContainer: [{
                type: ViewChild,
                args: ['chartContainer', { static: true }]
            }] } });
