// @lk/core/esm2022/lib/directives/tooltip.directive.mjs
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class TooltipDirective {
    constructor(elementRef, renderer) {
        this.elementRef = elementRef;
        this.renderer = renderer;
        this.appTooltip = '';
        this.tooltipPosition = 'below';
        this.tooltipShowDelay = 0;
        this.tooltipHideDelay = 0;
        this.tooltipDisabled = false;
    }
    ngOnInit() {
        // The tooltip will be handled by Angular Material's matTooltip directive
        // This directive is a wrapper that applies the tooltip functionality
        if (this.appTooltip && !this.tooltipDisabled) {
            this.renderer.setAttribute(this.elementRef.nativeElement, 'matTooltip', this.appTooltip);
            this.renderer.setAttribute(this.elementRef.nativeElement, 'matTooltipPosition', this.tooltipPosition);
            this.renderer.setAttribute(this.elementRef.nativeElement, 'matTooltipShowDelay', this.tooltipShowDelay.toString());
            this.renderer.setAttribute(this.elementRef.nativeElement, 'matTooltipHideDelay', this.tooltipHideDelay.toString());
        }
    }
    ngOnDestroy() {
        // Cleanup if needed
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TooltipDirective, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "17.3.12", type: TooltipDirective, isStandalone: true, selector: "[appTooltip]", inputs: { appTooltip: "appTooltip", tooltipPosition: "tooltipPosition", tooltipShowDelay: "tooltipShowDelay", tooltipHideDelay: "tooltipHideDelay", tooltipDisabled: "tooltipDisabled" }, ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: TooltipDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appTooltip]',
                    standalone: true
                }]
        }], ctorParameters: () => [{ type: i0.ElementRef }, { type: i0.Renderer2 }], propDecorators: { appTooltip: [{
                type: Input
            }], tooltipPosition: [{
                type: Input
            }], tooltipShowDelay: [{
                type: Input
            }], tooltipHideDelay: [{
                type: Input
            }], tooltipDisabled: [{
                type: Input
            }] } });
