// @lk/core/esm2022/lib/tools/checkbox/checkbox.component.mjs
import { CommonModule } from '@angular/common';
import { Component, Input, forwardRef } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { MatCheckboxModule } from '@angular/material/checkbox';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/material/checkbox";
export class CheckboxComponent {
    constructor() {
        this.label = '';
        this.disabled = false;
        this.required = false;
        /** Optional template to render instead of the plain label text */
        this.labelTemplateRef = null;
        this.value = false;
        this.onChange = (_) => { };
        this.onTouched = () => { };
    }
    writeValue(value) {
        this.value = value;
    }
    registerOnChange(fn) {
        this.onChange = fn;
    }
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    setDisabledState(isDisabled) {
        this.disabled = isDisabled;
    }
    onCheckboxChange(event) {
        this.value = event.checked;
        this.onChange(this.value);
        this.onTouched();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: CheckboxComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "17.0.0", version: "17.3.12", type: CheckboxComponent, isStandalone: true, selector: "app-checkbox", inputs: { label: "label", disabled: "disabled", required: "required", labelTemplateRef: "labelTemplateRef" }, providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                useExisting: forwardRef(() => CheckboxComponent),
                multi: true
            }
        ], ngImport: i0, template: "<mat-checkbox\n  [checked]=\"value\"\n  [disabled]=\"disabled\"\n  [required]=\"required\"\n  (change)=\"onCheckboxChange($event)\">\n  @if (labelTemplateRef) {\n    <ng-container *ngTemplateOutlet=\"labelTemplateRef; context: { $implicit: label, label: label }\"></ng-container>\n  } @else {\n    {{ label }}\n  }\n</mat-checkbox>", styles: [":host{display:inline-flex;align-items:center}mat-checkbox{font-size:15px}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "directive", type: i1.NgTemplateOutlet, selector: "[ngTemplateOutlet]", inputs: ["ngTemplateOutletContext", "ngTemplateOutlet", "ngTemplateOutletInjector"] }, { kind: "ngmodule", type: MatCheckboxModule }, { kind: "component", type: i2.MatCheckbox, selector: "mat-checkbox", inputs: ["aria-label", "aria-labelledby", "aria-describedby", "id", "required", "labelPosition", "name", "value", "disableRipple", "tabIndex", "color", "checked", "disabled", "indeterminate"], outputs: ["change", "indeterminateChange"], exportAs: ["matCheckbox"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: CheckboxComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-checkbox', standalone: true, imports: [CommonModule, MatCheckboxModule], providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef(() => CheckboxComponent),
                            multi: true
                        }
                    ], template: "<mat-checkbox\n  [checked]=\"value\"\n  [disabled]=\"disabled\"\n  [required]=\"required\"\n  (change)=\"onCheckboxChange($event)\">\n  @if (labelTemplateRef) {\n    <ng-container *ngTemplateOutlet=\"labelTemplateRef; context: { $implicit: label, label: label }\"></ng-container>\n  } @else {\n    {{ label }}\n  }\n</mat-checkbox>", styles: [":host{display:inline-flex;align-items:center}mat-checkbox{font-size:15px}\n"] }]
        }], propDecorators: { label: [{
                type: Input
            }], disabled: [{
                type: Input
            }], required: [{
                type: Input
            }], labelTemplateRef: [{
                type: Input
            }] } });
