// @lk/core/esm2022/lib/tools/image/image.component.mjs
import { Component, Input, Output, EventEmitter } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
export class ImageComponent {
    constructor() {
        this.src = undefined;
        this.alt = '';
        this.width = '';
        this.height = '';
        this.class = '';
        this.style = '';
        this.loading = 'lazy';
        this.decoding = 'async';
        this.crossorigin = undefined;
        this.config = undefined;
        this.clickable = false;
        this.imageLoad = new EventEmitter();
        this.imageError = new EventEmitter();
        this.imageClick = new EventEmitter();
        // Cached style to prevent repeated method calls
        this.cachedStyle = ''; // Public so template can access it directly
        this.lastWidth = '';
        this.lastHeight = '';
        this.lastStyle = '';
    }
    ngOnInit() {
        // If config is provided, override individual inputs
        if (this.config) {
            this.src = this.config.src || this.src;
            this.alt = this.config.alt || this.alt;
            this.width = this.config.width || this.width;
            this.height = this.config.height || this.height;
            this.class = this.config.class || this.class;
            this.style = this.config.style || this.style;
            this.loading = this.config.loading || this.loading;
            this.decoding = this.config.decoding || this.decoding;
            this.crossorigin = this.config.crossorigin || this.crossorigin;
        }
        this.updateCachedStyle();
    }
    ngOnChanges() {
        // Update cached style when inputs change
        if (this.width !== this.lastWidth || this.height !== this.lastHeight || this.style !== this.lastStyle) {
            this.updateCachedStyle();
            this.lastWidth = this.width;
            this.lastHeight = this.height;
            this.lastStyle = this.style;
        }
    }
    updateCachedStyle() {
        let style = this.style || '';
        if (this.width) {
            style += `width: ${typeof this.width === 'number' ? this.width + 'px' : this.width}; `;
        }
        if (this.height) {
            style += `height: ${typeof this.height === 'number' ? this.height + 'px' : this.height}; `;
        }
        this.cachedStyle = style.trim();
    }
    getImageStyle() {
        return this.cachedStyle;
    }
    onImageLoad(event) {
        this.imageLoad.emit(event);
    }
    onImageError(event) {
        this.imageError.emit(event);
    }
    onImageClick(event) {
        if (this.clickable) {
            this.imageClick.emit(event);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ImageComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: ImageComponent, isStandalone: true, selector: "app-image", inputs: { src: "src", alt: "alt", width: "width", height: "height", class: "class", style: "style", loading: "loading", decoding: "decoding", crossorigin: "crossorigin", config: "config", clickable: "clickable" }, outputs: { imageLoad: "imageLoad", imageError: "imageError", imageClick: "imageClick" }, usesOnChanges: true, ngImport: i0, template: "<img \n  [src]=\"src\" \n  [alt]=\"alt\"\n  [class]=\"class + (clickable ? ' clickable' : '')\"\n  [style]=\"cachedStyle\"\n  [loading]=\"loading\"\n  [decoding]=\"decoding\"\n  [attr.crossorigin]=\"crossorigin\"\n  (error)=\"onImageError($event)\"\n  (load)=\"onImageLoad($event)\"\n  (click)=\"onImageClick($event)\"\n  role=\"img\"\n  [attr.aria-label]=\"alt\"> ", styles: [":host{display:inline-block}img{max-width:100%;height:auto;display:block;border-radius:4px;transition:opacity .3s ease,transform .2s ease}img:hover{transform:scale(1.02)}img.rounded{border-radius:8px}img.circle{border-radius:50%}img.shadow{box-shadow:0 2px 8px #0000001a}img.border{border:1px solid #e0e0e0}img.clickable{cursor:pointer}img.clickable:hover{transform:scale(1.05);box-shadow:0 4px 12px #00000026}img.loading{opacity:.7}img.error{opacity:.5;filter:grayscale(100%)}.responsive{width:100%;height:auto}.contain{object-fit:contain}.cover{object-fit:cover}.fill{object-fit:fill}.small{max-width:100px}.medium{max-width:200px}.large{max-width:400px}.xlarge{max-width:600px}.aspect-ratio-16-9{position:relative;width:100%;height:0;padding-bottom:56.25%}.aspect-ratio-16-9 img{position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover}.aspect-ratio-4-3{position:relative;width:100%;height:0;padding-bottom:75%}.aspect-ratio-4-3 img{position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover}.aspect-ratio-1-1{position:relative;width:100%;height:0;padding-bottom:100%}.aspect-ratio-1-1 img{position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: ImageComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-image', standalone: true, imports: [CommonModule], template: "<img \n  [src]=\"src\" \n  [alt]=\"alt\"\n  [class]=\"class + (clickable ? ' clickable' : '')\"\n  [style]=\"cachedStyle\"\n  [loading]=\"loading\"\n  [decoding]=\"decoding\"\n  [attr.crossorigin]=\"crossorigin\"\n  (error)=\"onImageError($event)\"\n  (load)=\"onImageLoad($event)\"\n  (click)=\"onImageClick($event)\"\n  role=\"img\"\n  [attr.aria-label]=\"alt\"> ", styles: [":host{display:inline-block}img{max-width:100%;height:auto;display:block;border-radius:4px;transition:opacity .3s ease,transform .2s ease}img:hover{transform:scale(1.02)}img.rounded{border-radius:8px}img.circle{border-radius:50%}img.shadow{box-shadow:0 2px 8px #0000001a}img.border{border:1px solid #e0e0e0}img.clickable{cursor:pointer}img.clickable:hover{transform:scale(1.05);box-shadow:0 4px 12px #00000026}img.loading{opacity:.7}img.error{opacity:.5;filter:grayscale(100%)}.responsive{width:100%;height:auto}.contain{object-fit:contain}.cover{object-fit:cover}.fill{object-fit:fill}.small{max-width:100px}.medium{max-width:200px}.large{max-width:400px}.xlarge{max-width:600px}.aspect-ratio-16-9{position:relative;width:100%;height:0;padding-bottom:56.25%}.aspect-ratio-16-9 img{position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover}.aspect-ratio-4-3{position:relative;width:100%;height:0;padding-bottom:75%}.aspect-ratio-4-3 img{position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover}.aspect-ratio-1-1{position:relative;width:100%;height:0;padding-bottom:100%}.aspect-ratio-1-1 img{position:absolute;top:0;left:0;width:100%;height:100%;object-fit:cover}\n"] }]
        }], propDecorators: { src: [{
                type: Input
            }], alt: [{
                type: Input
            }], width: [{
                type: Input
            }], height: [{
                type: Input
            }], class: [{
                type: Input
            }], style: [{
                type: Input
            }], loading: [{
                type: Input
            }], decoding: [{
                type: Input
            }], crossorigin: [{
                type: Input
            }], config: [{
                type: Input
            }], clickable: [{
                type: Input
            }], imageLoad: [{
                type: Output
            }], imageError: [{
                type: Output
            }], imageClick: [{
                type: Output
            }] } });
