// @lk/core/esm2022/lib/tools/status-cell-renderer/status-cell-renderer.component.mjs
import { Component } from '@angular/core';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
export class StatusCellRendererComponent {
    constructor() {
        this.statusClass = 'status-chip status-neutral';
        this.displayValue = '';
    }
    agInit(params) {
        // value is required in ICellRendererParams, so it's always present
        const value = params.value ?? '';
        const field = params.statusField || params.colDef?.field;
        const mapping = params.statusMapping || this.getDefaultMapping(field);
        this.displayValue = value ? value.toString() : '';
        const valueStr = value ? value.toString().toLowerCase() : '';
        this.statusClass = `status-chip ${mapping[valueStr] || 'status-neutral'}`;
    }
    getDefaultMapping(field) {
        // Default mappings based on field name or common patterns
        if (field?.toLowerCase().includes('availability') || field?.toLowerCase().includes('available')) {
            return {
                'available': 'status-available',
                'unavailable': 'status-unavailable',
                'busy': 'status-busy',
                'offline': 'status-offline',
                'online': 'status-available',
                'active': 'status-available',
                'inactive': 'status-offline'
            };
        }
        // General status mapping
        return {
            'active': 'status-success',
            'inactive': 'status-neutral',
            'pending': 'status-warning',
            'approved': 'status-success',
            'rejected': 'status-danger',
            'cancelled': 'status-danger',
            'completed': 'status-success',
            'in progress': 'status-info',
            'draft': 'status-neutral',
            'published': 'status-success',
            'archived': 'status-neutral',
            'enabled': 'status-success',
            'disabled': 'status-neutral',
            'online': 'status-success',
            'offline': 'status-neutral',
            'available': 'status-available',
            'unavailable': 'status-unavailable',
            'busy': 'status-busy'
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: StatusCellRendererComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: StatusCellRendererComponent, isStandalone: true, selector: "app-status-cell-renderer", ngImport: i0, template: `
    <span [class]="statusClass">
      {{ displayValue }}
    </span>
  `, isInline: true, dependencies: [{ kind: "ngmodule", type: CommonModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: StatusCellRendererComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-status-cell-renderer', template: `
    <span [class]="statusClass">
      {{ displayValue }}
    </span>
  `, standalone: true, imports: [CommonModule] }]
        }] });
