// @lk/core/esm2022/lib/tools/stepper/stepper.component.mjs
import { Component, Input } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MatStepperModule } from '@angular/material/stepper';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
import * as i1 from "@angular/material/stepper";
export class StepperComponent {
    constructor() {
        this.orientation = 'horizontal';
        this.linear = false;
        this.selectedIndex = 0;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: StepperComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: StepperComponent, isStandalone: true, selector: "app-stepper", inputs: { orientation: "orientation", linear: "linear", selectedIndex: "selectedIndex" }, ngImport: i0, template: "<mat-stepper\n  [orientation]=\"orientation\"\n  [linear]=\"linear\"\n  [selectedIndex]=\"selectedIndex\"\n  class=\"app-stepper\"\n>\n  <ng-content></ng-content>\n</mat-stepper>\n\n", styles: [".app-stepper{width:100%}\n"], dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatStepperModule }, { kind: "component", type: i1.MatStepper, selector: "mat-stepper, mat-vertical-stepper, mat-horizontal-stepper, [matStepper]", inputs: ["disableRipple", "color", "labelPosition", "headerPosition", "animationDuration"], outputs: ["animationDone"], exportAs: ["matStepper", "matVerticalStepper", "matHorizontalStepper"] }, { kind: "ngmodule", type: FormsModule }, { kind: "ngmodule", type: ReactiveFormsModule }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: StepperComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-stepper', standalone: true, imports: [CommonModule, MatStepperModule, FormsModule, ReactiveFormsModule], template: "<mat-stepper\n  [orientation]=\"orientation\"\n  [linear]=\"linear\"\n  [selectedIndex]=\"selectedIndex\"\n  class=\"app-stepper\"\n>\n  <ng-content></ng-content>\n</mat-stepper>\n\n", styles: [".app-stepper{width:100%}\n"] }]
        }], propDecorators: { orientation: [{
                type: Input
            }], linear: [{
                type: Input
            }], selectedIndex: [{
                type: Input
            }] } });
export class StepComponent {
    constructor() {
        this.label = '';
        this.optional = false;
        this.completed = false;
        this.state = 'number';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: StepComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "17.3.12", type: StepComponent, isStandalone: true, selector: "app-step", inputs: { label: "label", optional: "optional", completed: "completed", state: "state" }, ngImport: i0, template: `
    <mat-step [label]="label" [optional]="optional" [completed]="completed" [state]="state">
      <ng-content></ng-content>
    </mat-step>
  `, isInline: true, dependencies: [{ kind: "ngmodule", type: CommonModule }, { kind: "ngmodule", type: MatStepperModule }, { kind: "component", type: i1.MatStep, selector: "mat-step", inputs: ["color"], exportAs: ["matStep"] }] }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: StepComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-step',
                    standalone: true,
                    imports: [CommonModule, MatStepperModule],
                    template: `
    <mat-step [label]="label" [optional]="optional" [completed]="completed" [state]="state">
      <ng-content></ng-content>
    </mat-step>
  `
                }]
        }], propDecorators: { label: [{
                type: Input
            }], optional: [{
                type: Input
            }], completed: [{
                type: Input
            }], state: [{
                type: Input
            }] } });
